// @polar-sh/sdk/dist/esm/models/errors/customernotready.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class CustomerNotReady extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "CustomerNotReady";
    }
}
/** @internal */
export const CustomerNotReady$inboundSchema = z.object({
    error: z.literal("CustomerNotReady"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new CustomerNotReady(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const CustomerNotReady$outboundSchema = z.instanceof(CustomerNotReady)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("CustomerNotReady"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerNotReady$;
(function (CustomerNotReady$) {
    /** @deprecated use `CustomerNotReady$inboundSchema` instead. */
    CustomerNotReady$.inboundSchema = CustomerNotReady$inboundSchema;
    /** @deprecated use `CustomerNotReady$outboundSchema` instead. */
    CustomerNotReady$.outboundSchema = CustomerNotReady$outboundSchema;
})(CustomerNotReady$ || (CustomerNotReady$ = {}));
