// @polar-sh/sdk/dist/commonjs/models/operations/benefitgrantslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantsListResponse$ = exports.BenefitGrantsListResponse$outboundSchema = exports.BenefitGrantsListResponse$inboundSchema = exports.BenefitGrantsListRequest$ = exports.BenefitGrantsListRequest$outboundSchema = exports.BenefitGrantsListRequest$inboundSchema = exports.BenefitGrantsListQueryParamCustomerIDFilter$ = exports.BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema = exports.BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema = exports.BenefitGrantsListQueryParamOrganizationIDFilter$ = exports.BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema = exports.BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.benefitGrantsListQueryParamOrganizationIDFilterToJSON = benefitGrantsListQueryParamOrganizationIDFilterToJSON;
exports.benefitGrantsListQueryParamOrganizationIDFilterFromJSON = benefitGrantsListQueryParamOrganizationIDFilterFromJSON;
exports.benefitGrantsListQueryParamCustomerIDFilterToJSON = benefitGrantsListQueryParamCustomerIDFilterToJSON;
exports.benefitGrantsListQueryParamCustomerIDFilterFromJSON = benefitGrantsListQueryParamCustomerIDFilterFromJSON;
exports.benefitGrantsListRequestToJSON = benefitGrantsListRequestToJSON;
exports.benefitGrantsListRequestFromJSON = benefitGrantsListRequestFromJSON;
exports.benefitGrantsListResponseToJSON = benefitGrantsListResponseToJSON;
exports.benefitGrantsListResponseFromJSON = benefitGrantsListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitgrantsortproperty_js_1 = require("../components/benefitgrantsortproperty.js");
const listresourcebenefitgrant_js_1 = require("../components/listresourcebenefitgrant.js");
/** @internal */
exports.BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantsListQueryParamOrganizationIDFilter$;
(function (BenefitGrantsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    BenefitGrantsListQueryParamOrganizationIDFilter$.inboundSchema = exports.BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    BenefitGrantsListQueryParamOrganizationIDFilter$.outboundSchema = exports.BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema;
})(BenefitGrantsListQueryParamOrganizationIDFilter$ || (exports.BenefitGrantsListQueryParamOrganizationIDFilter$ = BenefitGrantsListQueryParamOrganizationIDFilter$ = {}));
function benefitGrantsListQueryParamOrganizationIDFilterToJSON(benefitGrantsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema.parse(benefitGrantsListQueryParamOrganizationIDFilter));
}
function benefitGrantsListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantsListQueryParamCustomerIDFilter$;
(function (BenefitGrantsListQueryParamCustomerIDFilter$) {
    /** @deprecated use `BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema` instead. */
    BenefitGrantsListQueryParamCustomerIDFilter$.inboundSchema = exports.BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema;
    /** @deprecated use `BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema` instead. */
    BenefitGrantsListQueryParamCustomerIDFilter$.outboundSchema = exports.BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema;
})(BenefitGrantsListQueryParamCustomerIDFilter$ || (exports.BenefitGrantsListQueryParamCustomerIDFilter$ = BenefitGrantsListQueryParamCustomerIDFilter$ = {}));
function benefitGrantsListQueryParamCustomerIDFilterToJSON(benefitGrantsListQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema.parse(benefitGrantsListQueryParamCustomerIDFilter));
}
function benefitGrantsListQueryParamCustomerIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListQueryParamCustomerIDFilter' from JSON`);
}
/** @internal */
exports.BenefitGrantsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customer_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    is_granted: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(benefitgrantsortproperty_js_1.BenefitGrantSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "is_granted": "isGranted",
    });
});
/** @internal */
exports.BenefitGrantsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customerId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    isGranted: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(benefitgrantsortproperty_js_1.BenefitGrantSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        isGranted: "is_granted",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantsListRequest$;
(function (BenefitGrantsListRequest$) {
    /** @deprecated use `BenefitGrantsListRequest$inboundSchema` instead. */
    BenefitGrantsListRequest$.inboundSchema = exports.BenefitGrantsListRequest$inboundSchema;
    /** @deprecated use `BenefitGrantsListRequest$outboundSchema` instead. */
    BenefitGrantsListRequest$.outboundSchema = exports.BenefitGrantsListRequest$outboundSchema;
})(BenefitGrantsListRequest$ || (exports.BenefitGrantsListRequest$ = BenefitGrantsListRequest$ = {}));
function benefitGrantsListRequestToJSON(benefitGrantsListRequest) {
    return JSON.stringify(exports.BenefitGrantsListRequest$outboundSchema.parse(benefitGrantsListRequest));
}
function benefitGrantsListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListRequest' from JSON`);
}
/** @internal */
exports.BenefitGrantsListResponse$inboundSchema = z.object({
    Result: listresourcebenefitgrant_js_1.ListResourceBenefitGrant$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.BenefitGrantsListResponse$outboundSchema = z.object({
    result: listresourcebenefitgrant_js_1.ListResourceBenefitGrant$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantsListResponse$;
(function (BenefitGrantsListResponse$) {
    /** @deprecated use `BenefitGrantsListResponse$inboundSchema` instead. */
    BenefitGrantsListResponse$.inboundSchema = exports.BenefitGrantsListResponse$inboundSchema;
    /** @deprecated use `BenefitGrantsListResponse$outboundSchema` instead. */
    BenefitGrantsListResponse$.outboundSchema = exports.BenefitGrantsListResponse$outboundSchema;
})(BenefitGrantsListResponse$ || (exports.BenefitGrantsListResponse$ = BenefitGrantsListResponse$ = {}));
function benefitGrantsListResponseToJSON(benefitGrantsListResponse) {
    return JSON.stringify(exports.BenefitGrantsListResponse$outboundSchema.parse(benefitGrantsListResponse));
}
function benefitGrantsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListResponse' from JSON`);
}
