// @polar-sh/sdk/dist/commonjs/models/operations/customerportalseatsresendinvitation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalSeatsResendInvitationRequest$ = exports.CustomerPortalSeatsResendInvitationRequest$outboundSchema = exports.CustomerPortalSeatsResendInvitationRequest$inboundSchema = exports.CustomerPortalSeatsResendInvitationSecurity$ = exports.CustomerPortalSeatsResendInvitationSecurity$outboundSchema = exports.CustomerPortalSeatsResendInvitationSecurity$inboundSchema = void 0;
exports.customerPortalSeatsResendInvitationSecurityToJSON = customerPortalSeatsResendInvitationSecurityToJSON;
exports.customerPortalSeatsResendInvitationSecurityFromJSON = customerPortalSeatsResendInvitationSecurityFromJSON;
exports.customerPortalSeatsResendInvitationRequestToJSON = customerPortalSeatsResendInvitationRequestToJSON;
exports.customerPortalSeatsResendInvitationRequestFromJSON = customerPortalSeatsResendInvitationRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalSeatsResendInvitationSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalSeatsResendInvitationSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSeatsResendInvitationSecurity$;
(function (CustomerPortalSeatsResendInvitationSecurity$) {
    /** @deprecated use `CustomerPortalSeatsResendInvitationSecurity$inboundSchema` instead. */
    CustomerPortalSeatsResendInvitationSecurity$.inboundSchema = exports.CustomerPortalSeatsResendInvitationSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsResendInvitationSecurity$outboundSchema` instead. */
    CustomerPortalSeatsResendInvitationSecurity$.outboundSchema = exports.CustomerPortalSeatsResendInvitationSecurity$outboundSchema;
})(CustomerPortalSeatsResendInvitationSecurity$ || (exports.CustomerPortalSeatsResendInvitationSecurity$ = CustomerPortalSeatsResendInvitationSecurity$ = {}));
function customerPortalSeatsResendInvitationSecurityToJSON(customerPortalSeatsResendInvitationSecurity) {
    return JSON.stringify(exports.CustomerPortalSeatsResendInvitationSecurity$outboundSchema.parse(customerPortalSeatsResendInvitationSecurity));
}
function customerPortalSeatsResendInvitationSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSeatsResendInvitationSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsResendInvitationSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalSeatsResendInvitationRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
exports.CustomerPortalSeatsResendInvitationRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSeatsResendInvitationRequest$;
(function (CustomerPortalSeatsResendInvitationRequest$) {
    /** @deprecated use `CustomerPortalSeatsResendInvitationRequest$inboundSchema` instead. */
    CustomerPortalSeatsResendInvitationRequest$.inboundSchema = exports.CustomerPortalSeatsResendInvitationRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsResendInvitationRequest$outboundSchema` instead. */
    CustomerPortalSeatsResendInvitationRequest$.outboundSchema = exports.CustomerPortalSeatsResendInvitationRequest$outboundSchema;
})(CustomerPortalSeatsResendInvitationRequest$ || (exports.CustomerPortalSeatsResendInvitationRequest$ = CustomerPortalSeatsResendInvitationRequest$ = {}));
function customerPortalSeatsResendInvitationRequestToJSON(customerPortalSeatsResendInvitationRequest) {
    return JSON.stringify(exports.CustomerPortalSeatsResendInvitationRequest$outboundSchema.parse(customerPortalSeatsResendInvitationRequest));
}
function customerPortalSeatsResendInvitationRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSeatsResendInvitationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsResendInvitationRequest' from JSON`);
}
