// @polar-sh/sdk/dist/commonjs/models/operations/customerportalseatslistclaimedsubscriptions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$ = exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema = exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema = void 0;
exports.customerPortalSeatsListClaimedSubscriptionsSecurityToJSON = customerPortalSeatsListClaimedSubscriptionsSecurityToJSON;
exports.customerPortalSeatsListClaimedSubscriptionsSecurityFromJSON = customerPortalSeatsListClaimedSubscriptionsSecurityFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSeatsListClaimedSubscriptionsSecurity$;
(function (CustomerPortalSeatsListClaimedSubscriptionsSecurity$) {
    /** @deprecated use `CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema` instead. */
    CustomerPortalSeatsListClaimedSubscriptionsSecurity$.inboundSchema = exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema` instead. */
    CustomerPortalSeatsListClaimedSubscriptionsSecurity$.outboundSchema = exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema;
})(CustomerPortalSeatsListClaimedSubscriptionsSecurity$ || (exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$ = CustomerPortalSeatsListClaimedSubscriptionsSecurity$ = {}));
function customerPortalSeatsListClaimedSubscriptionsSecurityToJSON(customerPortalSeatsListClaimedSubscriptionsSecurity) {
    return JSON.stringify(exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema.parse(customerPortalSeatsListClaimedSubscriptionsSecurity));
}
function customerPortalSeatsListClaimedSubscriptionsSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsListClaimedSubscriptionsSecurity' from JSON`);
}
