// @polar-sh/sdk/dist/commonjs/models/operations/customerportalseatsrevokeseat.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalSeatsRevokeSeatRequest$ = exports.CustomerPortalSeatsRevokeSeatRequest$outboundSchema = exports.CustomerPortalSeatsRevokeSeatRequest$inboundSchema = exports.CustomerPortalSeatsRevokeSeatSecurity$ = exports.CustomerPortalSeatsRevokeSeatSecurity$outboundSchema = exports.CustomerPortalSeatsRevokeSeatSecurity$inboundSchema = void 0;
exports.customerPortalSeatsRevokeSeatSecurityToJSON = customerPortalSeatsRevokeSeatSecurityToJSON;
exports.customerPortalSeatsRevokeSeatSecurityFromJSON = customerPortalSeatsRevokeSeatSecurityFromJSON;
exports.customerPortalSeatsRevokeSeatRequestToJSON = customerPortalSeatsRevokeSeatRequestToJSON;
exports.customerPortalSeatsRevokeSeatRequestFromJSON = customerPortalSeatsRevokeSeatRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalSeatsRevokeSeatSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalSeatsRevokeSeatSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSeatsRevokeSeatSecurity$;
(function (CustomerPortalSeatsRevokeSeatSecurity$) {
    /** @deprecated use `CustomerPortalSeatsRevokeSeatSecurity$inboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatSecurity$.inboundSchema = exports.CustomerPortalSeatsRevokeSeatSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsRevokeSeatSecurity$outboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatSecurity$.outboundSchema = exports.CustomerPortalSeatsRevokeSeatSecurity$outboundSchema;
})(CustomerPortalSeatsRevokeSeatSecurity$ || (exports.CustomerPortalSeatsRevokeSeatSecurity$ = CustomerPortalSeatsRevokeSeatSecurity$ = {}));
function customerPortalSeatsRevokeSeatSecurityToJSON(customerPortalSeatsRevokeSeatSecurity) {
    return JSON.stringify(exports.CustomerPortalSeatsRevokeSeatSecurity$outboundSchema.parse(customerPortalSeatsRevokeSeatSecurity));
}
function customerPortalSeatsRevokeSeatSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSeatsRevokeSeatSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsRevokeSeatSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalSeatsRevokeSeatRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
exports.CustomerPortalSeatsRevokeSeatRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSeatsRevokeSeatRequest$;
(function (CustomerPortalSeatsRevokeSeatRequest$) {
    /** @deprecated use `CustomerPortalSeatsRevokeSeatRequest$inboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatRequest$.inboundSchema = exports.CustomerPortalSeatsRevokeSeatRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsRevokeSeatRequest$outboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatRequest$.outboundSchema = exports.CustomerPortalSeatsRevokeSeatRequest$outboundSchema;
})(CustomerPortalSeatsRevokeSeatRequest$ || (exports.CustomerPortalSeatsRevokeSeatRequest$ = CustomerPortalSeatsRevokeSeatRequest$ = {}));
function customerPortalSeatsRevokeSeatRequestToJSON(customerPortalSeatsRevokeSeatRequest) {
    return JSON.stringify(exports.CustomerPortalSeatsRevokeSeatRequest$outboundSchema.parse(customerPortalSeatsRevokeSeatRequest));
}
function customerPortalSeatsRevokeSeatRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSeatsRevokeSeatRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsRevokeSeatRequest' from JSON`);
}
