// @polar-sh/sdk/dist/commonjs/models/operations/oauth2clientsoauth2getclient.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth2ClientsOauth2GetClientRequest$ = exports.Oauth2ClientsOauth2GetClientRequest$outboundSchema = exports.Oauth2ClientsOauth2GetClientRequest$inboundSchema = void 0;
exports.oauth2ClientsOauth2GetClientRequestToJSON = oauth2ClientsOauth2GetClientRequestToJSON;
exports.oauth2ClientsOauth2GetClientRequestFromJSON = oauth2ClientsOauth2GetClientRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Oauth2ClientsOauth2GetClientRequest$inboundSchema = z.object({
    client_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_id": "clientId",
    });
});
/** @internal */
exports.Oauth2ClientsOauth2GetClientRequest$outboundSchema = z.object({
    clientId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientId: "client_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Oauth2ClientsOauth2GetClientRequest$;
(function (Oauth2ClientsOauth2GetClientRequest$) {
    /** @deprecated use `Oauth2ClientsOauth2GetClientRequest$inboundSchema` instead. */
    Oauth2ClientsOauth2GetClientRequest$.inboundSchema = exports.Oauth2ClientsOauth2GetClientRequest$inboundSchema;
    /** @deprecated use `Oauth2ClientsOauth2GetClientRequest$outboundSchema` instead. */
    Oauth2ClientsOauth2GetClientRequest$.outboundSchema = exports.Oauth2ClientsOauth2GetClientRequest$outboundSchema;
})(Oauth2ClientsOauth2GetClientRequest$ || (exports.Oauth2ClientsOauth2GetClientRequest$ = Oauth2ClientsOauth2GetClientRequest$ = {}));
function oauth2ClientsOauth2GetClientRequestToJSON(oauth2ClientsOauth2GetClientRequest) {
    return JSON.stringify(exports.Oauth2ClientsOauth2GetClientRequest$outboundSchema.parse(oauth2ClientsOauth2GetClientRequest));
}
function oauth2ClientsOauth2GetClientRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Oauth2ClientsOauth2GetClientRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Oauth2ClientsOauth2GetClientRequest' from JSON`);
}
