// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersExport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersExport = void 0;
const customersExport_js_1 = require("../../funcs/customersExport.js");
const customersexport_js_1 = require("../../models/operations/customersexport.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersexport_js_1.CustomersExportRequest$inboundSchema,
};
exports.tool$customersExport = {
    name: "customers-export",
    description: `Export Customers

Export customers as a CSV file.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersExport_js_1.customersExport)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
