// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersGetBalance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersGetBalance = void 0;
const customersGetBalance_js_1 = require("../../funcs/customersGetBalance.js");
const customersgetbalance_js_1 = require("../../models/operations/customersgetbalance.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersgetbalance_js_1.CustomersGetBalanceRequest$inboundSchema,
};
exports.tool$customersGetBalance = {
    name: "customers-get-balance",
    description: `Get Customer Balance

Get customer balance information.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersGetBalance_js_1.customersGetBalance)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
