// @polar-sh/sdk/dist/esm/models/operations/benefitgrantslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantSortProperty$inboundSchema, BenefitGrantSortProperty$outboundSchema, } from "../components/benefitgrantsortproperty.js";
import { ListResourceBenefitGrant$inboundSchema, ListResourceBenefitGrant$outboundSchema, } from "../components/listresourcebenefitgrant.js";
/** @internal */
export const BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantsListQueryParamOrganizationIDFilter$;
(function (BenefitGrantsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    BenefitGrantsListQueryParamOrganizationIDFilter$.inboundSchema = BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    BenefitGrantsListQueryParamOrganizationIDFilter$.outboundSchema = BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema;
})(BenefitGrantsListQueryParamOrganizationIDFilter$ || (BenefitGrantsListQueryParamOrganizationIDFilter$ = {}));
export function benefitGrantsListQueryParamOrganizationIDFilterToJSON(benefitGrantsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema.parse(benefitGrantsListQueryParamOrganizationIDFilter));
}
export function benefitGrantsListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantsListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
export const BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantsListQueryParamCustomerIDFilter$;
(function (BenefitGrantsListQueryParamCustomerIDFilter$) {
    /** @deprecated use `BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema` instead. */
    BenefitGrantsListQueryParamCustomerIDFilter$.inboundSchema = BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema;
    /** @deprecated use `BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema` instead. */
    BenefitGrantsListQueryParamCustomerIDFilter$.outboundSchema = BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema;
})(BenefitGrantsListQueryParamCustomerIDFilter$ || (BenefitGrantsListQueryParamCustomerIDFilter$ = {}));
export function benefitGrantsListQueryParamCustomerIDFilterToJSON(benefitGrantsListQueryParamCustomerIDFilter) {
    return JSON.stringify(BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema.parse(benefitGrantsListQueryParamCustomerIDFilter));
}
export function benefitGrantsListQueryParamCustomerIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantsListQueryParamCustomerIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListQueryParamCustomerIDFilter' from JSON`);
}
/** @internal */
export const BenefitGrantsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customer_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    is_granted: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(BenefitGrantSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "is_granted": "isGranted",
    });
});
/** @internal */
export const BenefitGrantsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customerId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    isGranted: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(BenefitGrantSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        isGranted: "is_granted",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantsListRequest$;
(function (BenefitGrantsListRequest$) {
    /** @deprecated use `BenefitGrantsListRequest$inboundSchema` instead. */
    BenefitGrantsListRequest$.inboundSchema = BenefitGrantsListRequest$inboundSchema;
    /** @deprecated use `BenefitGrantsListRequest$outboundSchema` instead. */
    BenefitGrantsListRequest$.outboundSchema = BenefitGrantsListRequest$outboundSchema;
})(BenefitGrantsListRequest$ || (BenefitGrantsListRequest$ = {}));
export function benefitGrantsListRequestToJSON(benefitGrantsListRequest) {
    return JSON.stringify(BenefitGrantsListRequest$outboundSchema.parse(benefitGrantsListRequest));
}
export function benefitGrantsListRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListRequest' from JSON`);
}
/** @internal */
export const BenefitGrantsListResponse$inboundSchema = z.object({
    Result: ListResourceBenefitGrant$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
/** @internal */
export const BenefitGrantsListResponse$outboundSchema = z.object({
    result: ListResourceBenefitGrant$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantsListResponse$;
(function (BenefitGrantsListResponse$) {
    /** @deprecated use `BenefitGrantsListResponse$inboundSchema` instead. */
    BenefitGrantsListResponse$.inboundSchema = BenefitGrantsListResponse$inboundSchema;
    /** @deprecated use `BenefitGrantsListResponse$outboundSchema` instead. */
    BenefitGrantsListResponse$.outboundSchema = BenefitGrantsListResponse$outboundSchema;
})(BenefitGrantsListResponse$ || (BenefitGrantsListResponse$ = {}));
export function benefitGrantsListResponseToJSON(benefitGrantsListResponse) {
    return JSON.stringify(BenefitGrantsListResponse$outboundSchema.parse(benefitGrantsListResponse));
}
export function benefitGrantsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListResponse' from JSON`);
}
