// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersessionintrospect.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalCustomerSessionIntrospectSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalCustomerSessionIntrospectSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomerSessionIntrospectSecurity$;
(function (CustomerPortalCustomerSessionIntrospectSecurity$) {
    /** @deprecated use `CustomerPortalCustomerSessionIntrospectSecurity$inboundSchema` instead. */
    CustomerPortalCustomerSessionIntrospectSecurity$.inboundSchema = CustomerPortalCustomerSessionIntrospectSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomerSessionIntrospectSecurity$outboundSchema` instead. */
    CustomerPortalCustomerSessionIntrospectSecurity$.outboundSchema = CustomerPortalCustomerSessionIntrospectSecurity$outboundSchema;
})(CustomerPortalCustomerSessionIntrospectSecurity$ || (CustomerPortalCustomerSessionIntrospectSecurity$ = {}));
export function customerPortalCustomerSessionIntrospectSecurityToJSON(customerPortalCustomerSessionIntrospectSecurity) {
    return JSON.stringify(CustomerPortalCustomerSessionIntrospectSecurity$outboundSchema.parse(customerPortalCustomerSessionIntrospectSecurity));
}
export function customerPortalCustomerSessionIntrospectSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomerSessionIntrospectSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomerSessionIntrospectSecurity' from JSON`);
}
