// @polar-sh/sdk/dist/esm/models/operations/customersexport.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomersExportQueryParamOrganizationId$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const CustomersExportQueryParamOrganizationId$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomersExportQueryParamOrganizationId$;
(function (CustomersExportQueryParamOrganizationId$) {
    /** @deprecated use `CustomersExportQueryParamOrganizationId$inboundSchema` instead. */
    CustomersExportQueryParamOrganizationId$.inboundSchema = CustomersExportQueryParamOrganizationId$inboundSchema;
    /** @deprecated use `CustomersExportQueryParamOrganizationId$outboundSchema` instead. */
    CustomersExportQueryParamOrganizationId$.outboundSchema = CustomersExportQueryParamOrganizationId$outboundSchema;
})(CustomersExportQueryParamOrganizationId$ || (CustomersExportQueryParamOrganizationId$ = {}));
export function customersExportQueryParamOrganizationIdToJSON(customersExportQueryParamOrganizationId) {
    return JSON.stringify(CustomersExportQueryParamOrganizationId$outboundSchema.parse(customersExportQueryParamOrganizationId));
}
export function customersExportQueryParamOrganizationIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersExportQueryParamOrganizationId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersExportQueryParamOrganizationId' from JSON`);
}
/** @internal */
export const CustomersExportRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CustomersExportRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomersExportRequest$;
(function (CustomersExportRequest$) {
    /** @deprecated use `CustomersExportRequest$inboundSchema` instead. */
    CustomersExportRequest$.inboundSchema = CustomersExportRequest$inboundSchema;
    /** @deprecated use `CustomersExportRequest$outboundSchema` instead. */
    CustomersExportRequest$.outboundSchema = CustomersExportRequest$outboundSchema;
})(CustomersExportRequest$ || (CustomersExportRequest$ = {}));
export function customersExportRequestToJSON(customersExportRequest) {
    return JSON.stringify(CustomersExportRequest$outboundSchema.parse(customersExportRequest));
}
export function customersExportRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersExportRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersExportRequest' from JSON`);
}
