// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersconfirmpaymentmethod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalCustomersConfirmPaymentMethodSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalCustomersConfirmPaymentMethodSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomersConfirmPaymentMethodSecurity$;
(function (CustomerPortalCustomersConfirmPaymentMethodSecurity$) {
    /** @deprecated use `CustomerPortalCustomersConfirmPaymentMethodSecurity$inboundSchema` instead. */
    CustomerPortalCustomersConfirmPaymentMethodSecurity$.inboundSchema = CustomerPortalCustomersConfirmPaymentMethodSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersConfirmPaymentMethodSecurity$outboundSchema` instead. */
    CustomerPortalCustomersConfirmPaymentMethodSecurity$.outboundSchema = CustomerPortalCustomersConfirmPaymentMethodSecurity$outboundSchema;
})(CustomerPortalCustomersConfirmPaymentMethodSecurity$ || (CustomerPortalCustomersConfirmPaymentMethodSecurity$ = {}));
export function customerPortalCustomersConfirmPaymentMethodSecurityToJSON(customerPortalCustomersConfirmPaymentMethodSecurity) {
    return JSON.stringify(CustomerPortalCustomersConfirmPaymentMethodSecurity$outboundSchema.parse(customerPortalCustomersConfirmPaymentMethodSecurity));
}
export function customerPortalCustomersConfirmPaymentMethodSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersConfirmPaymentMethodSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersConfirmPaymentMethodSecurity' from JSON`);
}
