// @polar-sh/sdk/dist/esm/models/operations/customerportalseatslistseats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSeatsListSeatsSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSeatsListSeatsSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsListSeatsSecurity$;
(function (CustomerPortalSeatsListSeatsSecurity$) {
    /** @deprecated use `CustomerPortalSeatsListSeatsSecurity$inboundSchema` instead. */
    CustomerPortalSeatsListSeatsSecurity$.inboundSchema = CustomerPortalSeatsListSeatsSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsListSeatsSecurity$outboundSchema` instead. */
    CustomerPortalSeatsListSeatsSecurity$.outboundSchema = CustomerPortalSeatsListSeatsSecurity$outboundSchema;
})(CustomerPortalSeatsListSeatsSecurity$ || (CustomerPortalSeatsListSeatsSecurity$ = {}));
export function customerPortalSeatsListSeatsSecurityToJSON(customerPortalSeatsListSeatsSecurity) {
    return JSON.stringify(CustomerPortalSeatsListSeatsSecurity$outboundSchema.parse(customerPortalSeatsListSeatsSecurity));
}
export function customerPortalSeatsListSeatsSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsListSeatsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsListSeatsSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalSeatsListSeatsRequest$inboundSchema = z.object({
    subscription_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
export const CustomerPortalSeatsListSeatsRequest$outboundSchema = z.object({
    subscriptionId: z.string(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsListSeatsRequest$;
(function (CustomerPortalSeatsListSeatsRequest$) {
    /** @deprecated use `CustomerPortalSeatsListSeatsRequest$inboundSchema` instead. */
    CustomerPortalSeatsListSeatsRequest$.inboundSchema = CustomerPortalSeatsListSeatsRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsListSeatsRequest$outboundSchema` instead. */
    CustomerPortalSeatsListSeatsRequest$.outboundSchema = CustomerPortalSeatsListSeatsRequest$outboundSchema;
})(CustomerPortalSeatsListSeatsRequest$ || (CustomerPortalSeatsListSeatsRequest$ = {}));
export function customerPortalSeatsListSeatsRequestToJSON(customerPortalSeatsListSeatsRequest) {
    return JSON.stringify(CustomerPortalSeatsListSeatsRequest$outboundSchema.parse(customerPortalSeatsListSeatsRequest));
}
export function customerPortalSeatsListSeatsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsListSeatsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsListSeatsRequest' from JSON`);
}
