// @polar-sh/sdk/dist/esm/models/operations/customersgetbalance.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomersGetBalanceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomersGetBalanceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomersGetBalanceRequest$;
(function (CustomersGetBalanceRequest$) {
    /** @deprecated use `CustomersGetBalanceRequest$inboundSchema` instead. */
    CustomersGetBalanceRequest$.inboundSchema = CustomersGetBalanceRequest$inboundSchema;
    /** @deprecated use `CustomersGetBalanceRequest$outboundSchema` instead. */
    CustomersGetBalanceRequest$.outboundSchema = CustomersGetBalanceRequest$outboundSchema;
})(CustomersGetBalanceRequest$ || (CustomersGetBalanceRequest$ = {}));
export function customersGetBalanceRequestToJSON(customersGetBalanceRequest) {
    return JSON.stringify(CustomersGetBalanceRequest$outboundSchema.parse(customersGetBalanceRequest));
}
export function customersGetBalanceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersGetBalanceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersGetBalanceRequest' from JSON`);
}
