// @polar-sh/sdk/dist/esm/models/operations/customerportalseatsresendinvitation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSeatsResendInvitationSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSeatsResendInvitationSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsResendInvitationSecurity$;
(function (CustomerPortalSeatsResendInvitationSecurity$) {
    /** @deprecated use `CustomerPortalSeatsResendInvitationSecurity$inboundSchema` instead. */
    CustomerPortalSeatsResendInvitationSecurity$.inboundSchema = CustomerPortalSeatsResendInvitationSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsResendInvitationSecurity$outboundSchema` instead. */
    CustomerPortalSeatsResendInvitationSecurity$.outboundSchema = CustomerPortalSeatsResendInvitationSecurity$outboundSchema;
})(CustomerPortalSeatsResendInvitationSecurity$ || (CustomerPortalSeatsResendInvitationSecurity$ = {}));
export function customerPortalSeatsResendInvitationSecurityToJSON(customerPortalSeatsResendInvitationSecurity) {
    return JSON.stringify(CustomerPortalSeatsResendInvitationSecurity$outboundSchema.parse(customerPortalSeatsResendInvitationSecurity));
}
export function customerPortalSeatsResendInvitationSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsResendInvitationSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsResendInvitationSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalSeatsResendInvitationRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
export const CustomerPortalSeatsResendInvitationRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsResendInvitationRequest$;
(function (CustomerPortalSeatsResendInvitationRequest$) {
    /** @deprecated use `CustomerPortalSeatsResendInvitationRequest$inboundSchema` instead. */
    CustomerPortalSeatsResendInvitationRequest$.inboundSchema = CustomerPortalSeatsResendInvitationRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsResendInvitationRequest$outboundSchema` instead. */
    CustomerPortalSeatsResendInvitationRequest$.outboundSchema = CustomerPortalSeatsResendInvitationRequest$outboundSchema;
})(CustomerPortalSeatsResendInvitationRequest$ || (CustomerPortalSeatsResendInvitationRequest$ = {}));
export function customerPortalSeatsResendInvitationRequestToJSON(customerPortalSeatsResendInvitationRequest) {
    return JSON.stringify(CustomerPortalSeatsResendInvitationRequest$outboundSchema.parse(customerPortalSeatsResendInvitationRequest));
}
export function customerPortalSeatsResendInvitationRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsResendInvitationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsResendInvitationRequest' from JSON`);
}
