// @polar-sh/sdk/dist/esm/models/operations/customerportalseatslistclaimedsubscriptions.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsListClaimedSubscriptionsSecurity$;
(function (CustomerPortalSeatsListClaimedSubscriptionsSecurity$) {
    /** @deprecated use `CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema` instead. */
    CustomerPortalSeatsListClaimedSubscriptionsSecurity$.inboundSchema = CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema` instead. */
    CustomerPortalSeatsListClaimedSubscriptionsSecurity$.outboundSchema = CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema;
})(CustomerPortalSeatsListClaimedSubscriptionsSecurity$ || (CustomerPortalSeatsListClaimedSubscriptionsSecurity$ = {}));
export function customerPortalSeatsListClaimedSubscriptionsSecurityToJSON(customerPortalSeatsListClaimedSubscriptionsSecurity) {
    return JSON.stringify(CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema.parse(customerPortalSeatsListClaimedSubscriptionsSecurity));
}
export function customerPortalSeatsListClaimedSubscriptionsSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsListClaimedSubscriptionsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsListClaimedSubscriptionsSecurity' from JSON`);
}
