// @polar-sh/sdk/dist/esm/models/operations/customerportalseatsrevokeseat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSeatsRevokeSeatSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSeatsRevokeSeatSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsRevokeSeatSecurity$;
(function (CustomerPortalSeatsRevokeSeatSecurity$) {
    /** @deprecated use `CustomerPortalSeatsRevokeSeatSecurity$inboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatSecurity$.inboundSchema = CustomerPortalSeatsRevokeSeatSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsRevokeSeatSecurity$outboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatSecurity$.outboundSchema = CustomerPortalSeatsRevokeSeatSecurity$outboundSchema;
})(CustomerPortalSeatsRevokeSeatSecurity$ || (CustomerPortalSeatsRevokeSeatSecurity$ = {}));
export function customerPortalSeatsRevokeSeatSecurityToJSON(customerPortalSeatsRevokeSeatSecurity) {
    return JSON.stringify(CustomerPortalSeatsRevokeSeatSecurity$outboundSchema.parse(customerPortalSeatsRevokeSeatSecurity));
}
export function customerPortalSeatsRevokeSeatSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsRevokeSeatSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsRevokeSeatSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalSeatsRevokeSeatRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
export const CustomerPortalSeatsRevokeSeatRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsRevokeSeatRequest$;
(function (CustomerPortalSeatsRevokeSeatRequest$) {
    /** @deprecated use `CustomerPortalSeatsRevokeSeatRequest$inboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatRequest$.inboundSchema = CustomerPortalSeatsRevokeSeatRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsRevokeSeatRequest$outboundSchema` instead. */
    CustomerPortalSeatsRevokeSeatRequest$.outboundSchema = CustomerPortalSeatsRevokeSeatRequest$outboundSchema;
})(CustomerPortalSeatsRevokeSeatRequest$ || (CustomerPortalSeatsRevokeSeatRequest$ = {}));
export function customerPortalSeatsRevokeSeatRequestToJSON(customerPortalSeatsRevokeSeatRequest) {
    return JSON.stringify(CustomerPortalSeatsRevokeSeatRequest$outboundSchema.parse(customerPortalSeatsRevokeSeatRequest));
}
export function customerPortalSeatsRevokeSeatRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsRevokeSeatRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsRevokeSeatRequest' from JSON`);
}
