// @polar-sh/sdk/dist/esm/models/operations/customerportalseatsassignseat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSeatsAssignSeatSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSeatsAssignSeatSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSeatsAssignSeatSecurity$;
(function (CustomerPortalSeatsAssignSeatSecurity$) {
    /** @deprecated use `CustomerPortalSeatsAssignSeatSecurity$inboundSchema` instead. */
    CustomerPortalSeatsAssignSeatSecurity$.inboundSchema = CustomerPortalSeatsAssignSeatSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSeatsAssignSeatSecurity$outboundSchema` instead. */
    CustomerPortalSeatsAssignSeatSecurity$.outboundSchema = CustomerPortalSeatsAssignSeatSecurity$outboundSchema;
})(CustomerPortalSeatsAssignSeatSecurity$ || (CustomerPortalSeatsAssignSeatSecurity$ = {}));
export function customerPortalSeatsAssignSeatSecurityToJSON(customerPortalSeatsAssignSeatSecurity) {
    return JSON.stringify(CustomerPortalSeatsAssignSeatSecurity$outboundSchema.parse(customerPortalSeatsAssignSeatSecurity));
}
export function customerPortalSeatsAssignSeatSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsAssignSeatSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsAssignSeatSecurity' from JSON`);
}
