// @polar-sh/sdk/dist/esm/models/operations/oauth2clientsoauth2getclient.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const Oauth2ClientsOauth2GetClientRequest$inboundSchema = z.object({
    client_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "client_id": "clientId",
    });
});
/** @internal */
export const Oauth2ClientsOauth2GetClientRequest$outboundSchema = z.object({
    clientId: z.string(),
}).transform((v) => {
    return remap$(v, {
        clientId: "client_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Oauth2ClientsOauth2GetClientRequest$;
(function (Oauth2ClientsOauth2GetClientRequest$) {
    /** @deprecated use `Oauth2ClientsOauth2GetClientRequest$inboundSchema` instead. */
    Oauth2ClientsOauth2GetClientRequest$.inboundSchema = Oauth2ClientsOauth2GetClientRequest$inboundSchema;
    /** @deprecated use `Oauth2ClientsOauth2GetClientRequest$outboundSchema` instead. */
    Oauth2ClientsOauth2GetClientRequest$.outboundSchema = Oauth2ClientsOauth2GetClientRequest$outboundSchema;
})(Oauth2ClientsOauth2GetClientRequest$ || (Oauth2ClientsOauth2GetClientRequest$ = {}));
export function oauth2ClientsOauth2GetClientRequestToJSON(oauth2ClientsOauth2GetClientRequest) {
    return JSON.stringify(Oauth2ClientsOauth2GetClientRequest$outboundSchema.parse(oauth2ClientsOauth2GetClientRequest));
}
export function oauth2ClientsOauth2GetClientRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Oauth2ClientsOauth2GetClientRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Oauth2ClientsOauth2GetClientRequest' from JSON`);
}
