// @polar-sh/sdk/dist/esm/models/operations/oauth2clientsoauth2deleteclient.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const Oauth2ClientsOauth2DeleteClientRequest$inboundSchema = z.object({
    client_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "client_id": "clientId",
    });
});
/** @internal */
export const Oauth2ClientsOauth2DeleteClientRequest$outboundSchema = z.object({
    clientId: z.string(),
}).transform((v) => {
    return remap$(v, {
        clientId: "client_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Oauth2ClientsOauth2DeleteClientRequest$;
(function (Oauth2ClientsOauth2DeleteClientRequest$) {
    /** @deprecated use `Oauth2ClientsOauth2DeleteClientRequest$inboundSchema` instead. */
    Oauth2ClientsOauth2DeleteClientRequest$.inboundSchema = Oauth2ClientsOauth2DeleteClientRequest$inboundSchema;
    /** @deprecated use `Oauth2ClientsOauth2DeleteClientRequest$outboundSchema` instead. */
    Oauth2ClientsOauth2DeleteClientRequest$.outboundSchema = Oauth2ClientsOauth2DeleteClientRequest$outboundSchema;
})(Oauth2ClientsOauth2DeleteClientRequest$ || (Oauth2ClientsOauth2DeleteClientRequest$ = {}));
export function oauth2ClientsOauth2DeleteClientRequestToJSON(oauth2ClientsOauth2DeleteClientRequest) {
    return JSON.stringify(Oauth2ClientsOauth2DeleteClientRequest$outboundSchema.parse(oauth2ClientsOauth2DeleteClientRequest));
}
export function oauth2ClientsOauth2DeleteClientRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Oauth2ClientsOauth2DeleteClientRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Oauth2ClientsOauth2DeleteClientRequest' from JSON`);
}
