// @polar-sh/sdk/dist/commonjs/models/components/address.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Address$ = exports.Address$outboundSchema = exports.Address$inboundSchema = exports.CountryAlpha2$ = exports.CountryAlpha2$outboundSchema = exports.CountryAlpha2$inboundSchema = exports.CountryAlpha2 = void 0;
exports.addressToJSON = addressToJSON;
exports.addressFromJSON = addressFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.CountryAlpha2 = {
    Ad: "AD",
    Ae: "AE",
    Af: "AF",
    Ag: "AG",
    Ai: "AI",
    Al: "AL",
    Am: "AM",
    Ao: "AO",
    Aq: "AQ",
    Ar: "AR",
    As: "AS",
    At: "AT",
    Au: "AU",
    Aw: "AW",
    Ax: "AX",
    Az: "AZ",
    Ba: "BA",
    Bb: "BB",
    Bd: "BD",
    Be: "BE",
    Bf: "BF",
    Bg: "BG",
    Bh: "BH",
    Bi: "BI",
    Bj: "BJ",
    Bl: "BL",
    Bm: "BM",
    Bn: "BN",
    Bo: "BO",
    Bq: "BQ",
    Br: "BR",
    Bs: "BS",
    Bt: "BT",
    Bv: "BV",
    Bw: "BW",
    By: "BY",
    Bz: "BZ",
    Ca: "CA",
    Cc: "CC",
    Cd: "CD",
    Cf: "CF",
    Cg: "CG",
    Ch: "CH",
    Ci: "CI",
    Ck: "CK",
    Cl: "CL",
    Cm: "CM",
    Cn: "CN",
    Co: "CO",
    Cr: "CR",
    Cu: "CU",
    Cv: "CV",
    Cw: "CW",
    Cx: "CX",
    Cy: "CY",
    Cz: "CZ",
    De: "DE",
    Dj: "DJ",
    Dk: "DK",
    Dm: "DM",
    Do: "DO",
    Dz: "DZ",
    Ec: "EC",
    Ee: "EE",
    Eg: "EG",
    Eh: "EH",
    Er: "ER",
    Es: "ES",
    Et: "ET",
    Fi: "FI",
    Fj: "FJ",
    Fk: "FK",
    Fm: "FM",
    Fo: "FO",
    Fr: "FR",
    Ga: "GA",
    Gb: "GB",
    Gd: "GD",
    Ge: "GE",
    Gf: "GF",
    Gg: "GG",
    Gh: "GH",
    Gi: "GI",
    Gl: "GL",
    Gm: "GM",
    Gn: "GN",
    Gp: "GP",
    Gq: "GQ",
    Gr: "GR",
    Gs: "GS",
    Gt: "GT",
    Gu: "GU",
    Gw: "GW",
    Gy: "GY",
    Hk: "HK",
    Hm: "HM",
    Hn: "HN",
    Hr: "HR",
    Ht: "HT",
    Hu: "HU",
    Id: "ID",
    Ie: "IE",
    Il: "IL",
    Im: "IM",
    In: "IN",
    Io: "IO",
    Iq: "IQ",
    Ir: "IR",
    Is: "IS",
    It: "IT",
    Je: "JE",
    Jm: "JM",
    Jo: "JO",
    Jp: "JP",
    Ke: "KE",
    Kg: "KG",
    Kh: "KH",
    Ki: "KI",
    Km: "KM",
    Kn: "KN",
    Kp: "KP",
    Kr: "KR",
    Kw: "KW",
    Ky: "KY",
    Kz: "KZ",
    La: "LA",
    Lb: "LB",
    Lc: "LC",
    Li: "LI",
    Lk: "LK",
    Lr: "LR",
    Ls: "LS",
    Lt: "LT",
    Lu: "LU",
    Lv: "LV",
    Ly: "LY",
    Ma: "MA",
    Mc: "MC",
    Md: "MD",
    Me: "ME",
    Mf: "MF",
    Mg: "MG",
    Mh: "MH",
    Mk: "MK",
    Ml: "ML",
    Mm: "MM",
    Mn: "MN",
    Mo: "MO",
    Mp: "MP",
    Mq: "MQ",
    Mr: "MR",
    Ms: "MS",
    Mt: "MT",
    Mu: "MU",
    Mv: "MV",
    Mw: "MW",
    Mx: "MX",
    My: "MY",
    Mz: "MZ",
    Na: "NA",
    Nc: "NC",
    Ne: "NE",
    Nf: "NF",
    Ng: "NG",
    Ni: "NI",
    Nl: "NL",
    No: "NO",
    Np: "NP",
    Nr: "NR",
    Nu: "NU",
    Nz: "NZ",
    Om: "OM",
    Pa: "PA",
    Pe: "PE",
    Pf: "PF",
    Pg: "PG",
    Ph: "PH",
    Pk: "PK",
    Pl: "PL",
    Pm: "PM",
    Pn: "PN",
    Pr: "PR",
    Ps: "PS",
    Pt: "PT",
    Pw: "PW",
    Py: "PY",
    Qa: "QA",
    Re: "RE",
    Ro: "RO",
    Rs: "RS",
    Ru: "RU",
    Rw: "RW",
    Sa: "SA",
    Sb: "SB",
    Sc: "SC",
    Sd: "SD",
    Se: "SE",
    Sg: "SG",
    Sh: "SH",
    Si: "SI",
    Sj: "SJ",
    Sk: "SK",
    Sl: "SL",
    Sm: "SM",
    Sn: "SN",
    So: "SO",
    Sr: "SR",
    Ss: "SS",
    St: "ST",
    Sv: "SV",
    Sx: "SX",
    Sy: "SY",
    Sz: "SZ",
    Tc: "TC",
    Td: "TD",
    Tf: "TF",
    Tg: "TG",
    Th: "TH",
    Tj: "TJ",
    Tk: "TK",
    Tl: "TL",
    Tm: "TM",
    Tn: "TN",
    To: "TO",
    Tr: "TR",
    Tt: "TT",
    Tv: "TV",
    Tw: "TW",
    Tz: "TZ",
    Ua: "UA",
    Ug: "UG",
    Um: "UM",
    Us: "US",
    Uy: "UY",
    Uz: "UZ",
    Va: "VA",
    Vc: "VC",
    Ve: "VE",
    Vg: "VG",
    Vi: "VI",
    Vn: "VN",
    Vu: "VU",
    Wf: "WF",
    Ws: "WS",
    Ye: "YE",
    Yt: "YT",
    Za: "ZA",
    Zm: "ZM",
    Zw: "ZW",
};
/** @internal */
exports.CountryAlpha2$inboundSchema = z.nativeEnum(exports.CountryAlpha2);
/** @internal */
exports.CountryAlpha2$outboundSchema = exports.CountryAlpha2$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CountryAlpha2$;
(function (CountryAlpha2$) {
    /** @deprecated use `CountryAlpha2$inboundSchema` instead. */
    CountryAlpha2$.inboundSchema = exports.CountryAlpha2$inboundSchema;
    /** @deprecated use `CountryAlpha2$outboundSchema` instead. */
    CountryAlpha2$.outboundSchema = exports.CountryAlpha2$outboundSchema;
})(CountryAlpha2$ || (exports.CountryAlpha2$ = CountryAlpha2$ = {}));
/** @internal */
exports.Address$inboundSchema = z.object({
    line1: z.nullable(z.string()).optional(),
    line2: z.nullable(z.string()).optional(),
    postal_code: z.nullable(z.string()).optional(),
    city: z.nullable(z.string()).optional(),
    state: z.nullable(z.string()).optional(),
    country: exports.CountryAlpha2$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "postal_code": "postalCode",
    });
});
/** @internal */
exports.Address$outboundSchema = z.object({
    line1: z.nullable(z.string()).optional(),
    line2: z.nullable(z.string()).optional(),
    postalCode: z.nullable(z.string()).optional(),
    city: z.nullable(z.string()).optional(),
    state: z.nullable(z.string()).optional(),
    country: exports.CountryAlpha2$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        postalCode: "postal_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Address$;
(function (Address$) {
    /** @deprecated use `Address$inboundSchema` instead. */
    Address$.inboundSchema = exports.Address$inboundSchema;
    /** @deprecated use `Address$outboundSchema` instead. */
    Address$.outboundSchema = exports.Address$outboundSchema;
})(Address$ || (exports.Address$ = Address$ = {}));
function addressToJSON(address) {
    return JSON.stringify(exports.Address$outboundSchema.parse(address));
}
function addressFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Address$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Address' from JSON`);
}
