// @polar-sh/sdk/dist/commonjs/models/components/benefitgrantsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantSortProperty$ = exports.BenefitGrantSortProperty$outboundSchema = exports.BenefitGrantSortProperty$inboundSchema = exports.BenefitGrantSortProperty = void 0;
const z = __importStar(require("zod"));
exports.BenefitGrantSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    GrantedAt: "granted_at",
    MinusGrantedAt: "-granted_at",
    RevokedAt: "revoked_at",
    MinusRevokedAt: "-revoked_at",
};
/** @internal */
exports.BenefitGrantSortProperty$inboundSchema = z.nativeEnum(exports.BenefitGrantSortProperty);
/** @internal */
exports.BenefitGrantSortProperty$outboundSchema = exports.BenefitGrantSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantSortProperty$;
(function (BenefitGrantSortProperty$) {
    /** @deprecated use `BenefitGrantSortProperty$inboundSchema` instead. */
    BenefitGrantSortProperty$.inboundSchema = exports.BenefitGrantSortProperty$inboundSchema;
    /** @deprecated use `BenefitGrantSortProperty$outboundSchema` instead. */
    BenefitGrantSortProperty$.outboundSchema = exports.BenefitGrantSortProperty$outboundSchema;
})(BenefitGrantSortProperty$ || (exports.BenefitGrantSortProperty$ = BenefitGrantSortProperty$ = {}));
