// @polar-sh/sdk/dist/commonjs/models/components/customerpaymentmethodconfirm.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPaymentMethodConfirm$ = exports.CustomerPaymentMethodConfirm$outboundSchema = exports.CustomerPaymentMethodConfirm$inboundSchema = void 0;
exports.customerPaymentMethodConfirmToJSON = customerPaymentMethodConfirmToJSON;
exports.customerPaymentMethodConfirmFromJSON = customerPaymentMethodConfirmFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPaymentMethodConfirm$inboundSchema = z.object({
    setup_intent_id: z.string(),
    set_default: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "setup_intent_id": "setupIntentId",
        "set_default": "setDefault",
    });
});
/** @internal */
exports.CustomerPaymentMethodConfirm$outboundSchema = z.object({
    setupIntentId: z.string(),
    setDefault: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        setupIntentId: "setup_intent_id",
        setDefault: "set_default",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPaymentMethodConfirm$;
(function (CustomerPaymentMethodConfirm$) {
    /** @deprecated use `CustomerPaymentMethodConfirm$inboundSchema` instead. */
    CustomerPaymentMethodConfirm$.inboundSchema = exports.CustomerPaymentMethodConfirm$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodConfirm$outboundSchema` instead. */
    CustomerPaymentMethodConfirm$.outboundSchema = exports.CustomerPaymentMethodConfirm$outboundSchema;
})(CustomerPaymentMethodConfirm$ || (exports.CustomerPaymentMethodConfirm$ = CustomerPaymentMethodConfirm$ = {}));
function customerPaymentMethodConfirmToJSON(customerPaymentMethodConfirm) {
    return JSON.stringify(exports.CustomerPaymentMethodConfirm$outboundSchema.parse(customerPaymentMethodConfirm));
}
function customerPaymentMethodConfirmFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPaymentMethodConfirm$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodConfirm' from JSON`);
}
