// @polar-sh/sdk/dist/commonjs/models/components/customerseat.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSeat$ = exports.CustomerSeat$outboundSchema = exports.CustomerSeat$inboundSchema = void 0;
exports.customerSeatToJSON = customerSeatToJSON;
exports.customerSeatFromJSON = customerSeatFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const seatstatus_js_1 = require("./seatstatus.js");
/** @internal */
exports.CustomerSeat$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    subscription_id: z.string(),
    status: seatstatus_js_1.SeatStatus$inboundSchema,
    customer_id: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    invitation_token_expires_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    claimed_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    seat_metadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subscription_id": "subscriptionId",
        "customer_id": "customerId",
        "customer_email": "customerEmail",
        "invitation_token_expires_at": "invitationTokenExpiresAt",
        "claimed_at": "claimedAt",
        "revoked_at": "revokedAt",
        "seat_metadata": "seatMetadata",
    });
});
/** @internal */
exports.CustomerSeat$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    subscriptionId: z.string(),
    status: seatstatus_js_1.SeatStatus$outboundSchema,
    customerId: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    invitationTokenExpiresAt: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
    claimedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    seatMetadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subscriptionId: "subscription_id",
        customerId: "customer_id",
        customerEmail: "customer_email",
        invitationTokenExpiresAt: "invitation_token_expires_at",
        claimedAt: "claimed_at",
        revokedAt: "revoked_at",
        seatMetadata: "seat_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerSeat$;
(function (CustomerSeat$) {
    /** @deprecated use `CustomerSeat$inboundSchema` instead. */
    CustomerSeat$.inboundSchema = exports.CustomerSeat$inboundSchema;
    /** @deprecated use `CustomerSeat$outboundSchema` instead. */
    CustomerSeat$.outboundSchema = exports.CustomerSeat$outboundSchema;
})(CustomerSeat$ || (exports.CustomerSeat$ = CustomerSeat$ = {}));
function customerSeatToJSON(customerSeat) {
    return JSON.stringify(exports.CustomerSeat$outboundSchema.parse(customerSeat));
}
function customerSeatFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerSeat$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeat' from JSON`);
}
