// @polar-sh/sdk/dist/commonjs/models/components/oauth2clientconfiguration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth2ClientConfiguration$ = exports.OAuth2ClientConfiguration$outboundSchema = exports.OAuth2ClientConfiguration$inboundSchema = exports.GrantTypes$ = exports.GrantTypes$outboundSchema = exports.GrantTypes$inboundSchema = exports.TokenEndpointAuthMethod$ = exports.TokenEndpointAuthMethod$outboundSchema = exports.TokenEndpointAuthMethod$inboundSchema = exports.GrantTypes = exports.TokenEndpointAuthMethod = void 0;
exports.oAuth2ClientConfigurationToJSON = oAuth2ClientConfigurationToJSON;
exports.oAuth2ClientConfigurationFromJSON = oAuth2ClientConfigurationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.TokenEndpointAuthMethod = {
    ClientSecretBasic: "client_secret_basic",
    ClientSecretPost: "client_secret_post",
    None: "none",
};
exports.GrantTypes = {
    AuthorizationCode: "authorization_code",
    RefreshToken: "refresh_token",
};
/** @internal */
exports.TokenEndpointAuthMethod$inboundSchema = z.nativeEnum(exports.TokenEndpointAuthMethod);
/** @internal */
exports.TokenEndpointAuthMethod$outboundSchema = exports.TokenEndpointAuthMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TokenEndpointAuthMethod$;
(function (TokenEndpointAuthMethod$) {
    /** @deprecated use `TokenEndpointAuthMethod$inboundSchema` instead. */
    TokenEndpointAuthMethod$.inboundSchema = exports.TokenEndpointAuthMethod$inboundSchema;
    /** @deprecated use `TokenEndpointAuthMethod$outboundSchema` instead. */
    TokenEndpointAuthMethod$.outboundSchema = exports.TokenEndpointAuthMethod$outboundSchema;
})(TokenEndpointAuthMethod$ || (exports.TokenEndpointAuthMethod$ = TokenEndpointAuthMethod$ = {}));
/** @internal */
exports.GrantTypes$inboundSchema = z
    .nativeEnum(exports.GrantTypes);
/** @internal */
exports.GrantTypes$outboundSchema = exports.GrantTypes$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GrantTypes$;
(function (GrantTypes$) {
    /** @deprecated use `GrantTypes$inboundSchema` instead. */
    GrantTypes$.inboundSchema = exports.GrantTypes$inboundSchema;
    /** @deprecated use `GrantTypes$outboundSchema` instead. */
    GrantTypes$.outboundSchema = exports.GrantTypes$outboundSchema;
})(GrantTypes$ || (exports.GrantTypes$ = GrantTypes$ = {}));
/** @internal */
exports.OAuth2ClientConfiguration$inboundSchema = z.object({
    redirect_uris: z.array(z.string()),
    token_endpoint_auth_method: exports.TokenEndpointAuthMethod$inboundSchema.default("client_secret_post"),
    grant_types: z.array(exports.GrantTypes$inboundSchema).optional(),
    response_types: z.array(z.string()).optional(),
    scope: z.string().default("openid profile email user:read organizations:read organizations:write custom_fields:read custom_fields:write discounts:read discounts:write checkout_links:read checkout_links:write checkouts:read checkouts:write transactions:read transactions:write payouts:read payouts:write products:read products:write benefits:read benefits:write events:read events:write meters:read meters:write files:read files:write subscriptions:read subscriptions:write customers:read customers:write customer_meters:read customer_sessions:write orders:read orders:write refunds:read refunds:write payments:read metrics:read webhooks:read webhooks:write external_organizations:read license_keys:read license_keys:write repositories:read repositories:write issues:read issues:write customer_portal:read customer_portal:write notifications:read notifications:write notification_recipients:read notification_recipients:write"),
    client_name: z.string(),
    client_uri: z.nullable(z.string()).optional(),
    logo_uri: z.nullable(z.string()).optional(),
    tos_uri: z.nullable(z.string()).optional(),
    policy_uri: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "redirect_uris": "redirectUris",
        "token_endpoint_auth_method": "tokenEndpointAuthMethod",
        "grant_types": "grantTypes",
        "response_types": "responseTypes",
        "client_name": "clientName",
        "client_uri": "clientUri",
        "logo_uri": "logoUri",
        "tos_uri": "tosUri",
        "policy_uri": "policyUri",
    });
});
/** @internal */
exports.OAuth2ClientConfiguration$outboundSchema = z.object({
    redirectUris: z.array(z.string()),
    tokenEndpointAuthMethod: exports.TokenEndpointAuthMethod$outboundSchema.default("client_secret_post"),
    grantTypes: z.array(exports.GrantTypes$outboundSchema).optional(),
    responseTypes: z.array(z.string()).optional(),
    scope: z.string().default("openid profile email user:read organizations:read organizations:write custom_fields:read custom_fields:write discounts:read discounts:write checkout_links:read checkout_links:write checkouts:read checkouts:write transactions:read transactions:write payouts:read payouts:write products:read products:write benefits:read benefits:write events:read events:write meters:read meters:write files:read files:write subscriptions:read subscriptions:write customers:read customers:write customer_meters:read customer_sessions:write orders:read orders:write refunds:read refunds:write payments:read metrics:read webhooks:read webhooks:write external_organizations:read license_keys:read license_keys:write repositories:read repositories:write issues:read issues:write customer_portal:read customer_portal:write notifications:read notifications:write notification_recipients:read notification_recipients:write"),
    clientName: z.string(),
    clientUri: z.nullable(z.string()).optional(),
    logoUri: z.nullable(z.string()).optional(),
    tosUri: z.nullable(z.string()).optional(),
    policyUri: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        redirectUris: "redirect_uris",
        tokenEndpointAuthMethod: "token_endpoint_auth_method",
        grantTypes: "grant_types",
        responseTypes: "response_types",
        clientName: "client_name",
        clientUri: "client_uri",
        logoUri: "logo_uri",
        tosUri: "tos_uri",
        policyUri: "policy_uri",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OAuth2ClientConfiguration$;
(function (OAuth2ClientConfiguration$) {
    /** @deprecated use `OAuth2ClientConfiguration$inboundSchema` instead. */
    OAuth2ClientConfiguration$.inboundSchema = exports.OAuth2ClientConfiguration$inboundSchema;
    /** @deprecated use `OAuth2ClientConfiguration$outboundSchema` instead. */
    OAuth2ClientConfiguration$.outboundSchema = exports.OAuth2ClientConfiguration$outboundSchema;
})(OAuth2ClientConfiguration$ || (exports.OAuth2ClientConfiguration$ = OAuth2ClientConfiguration$ = {}));
function oAuth2ClientConfigurationToJSON(oAuth2ClientConfiguration) {
    return JSON.stringify(exports.OAuth2ClientConfiguration$outboundSchema.parse(oAuth2ClientConfiguration));
}
function oAuth2ClientConfigurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OAuth2ClientConfiguration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OAuth2ClientConfiguration' from JSON`);
}
