// @polar-sh/sdk/dist/commonjs/models/components/productcreaterecurring.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCreateRecurring$ = exports.ProductCreateRecurring$outboundSchema = exports.ProductCreateRecurring$inboundSchema = exports.ProductCreateRecurringPrices$ = exports.ProductCreateRecurringPrices$outboundSchema = exports.ProductCreateRecurringPrices$inboundSchema = exports.ProductCreateRecurringMetadata$ = exports.ProductCreateRecurringMetadata$outboundSchema = exports.ProductCreateRecurringMetadata$inboundSchema = void 0;
exports.productCreateRecurringMetadataToJSON = productCreateRecurringMetadataToJSON;
exports.productCreateRecurringMetadataFromJSON = productCreateRecurringMetadataFromJSON;
exports.productCreateRecurringPricesToJSON = productCreateRecurringPricesToJSON;
exports.productCreateRecurringPricesFromJSON = productCreateRecurringPricesFromJSON;
exports.productCreateRecurringToJSON = productCreateRecurringToJSON;
exports.productCreateRecurringFromJSON = productCreateRecurringFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const productpriceseatbasedcreate_js_1 = require("./productpriceseatbasedcreate.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.ProductCreateRecurringMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.ProductCreateRecurringMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateRecurringMetadata$;
(function (ProductCreateRecurringMetadata$) {
    /** @deprecated use `ProductCreateRecurringMetadata$inboundSchema` instead. */
    ProductCreateRecurringMetadata$.inboundSchema = exports.ProductCreateRecurringMetadata$inboundSchema;
    /** @deprecated use `ProductCreateRecurringMetadata$outboundSchema` instead. */
    ProductCreateRecurringMetadata$.outboundSchema = exports.ProductCreateRecurringMetadata$outboundSchema;
})(ProductCreateRecurringMetadata$ || (exports.ProductCreateRecurringMetadata$ = ProductCreateRecurringMetadata$ = {}));
function productCreateRecurringMetadataToJSON(productCreateRecurringMetadata) {
    return JSON.stringify(exports.ProductCreateRecurringMetadata$outboundSchema.parse(productCreateRecurringMetadata));
}
function productCreateRecurringMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateRecurringMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurringMetadata' from JSON`);
}
/** @internal */
exports.ProductCreateRecurringPrices$inboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
exports.ProductCreateRecurringPrices$outboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateRecurringPrices$;
(function (ProductCreateRecurringPrices$) {
    /** @deprecated use `ProductCreateRecurringPrices$inboundSchema` instead. */
    ProductCreateRecurringPrices$.inboundSchema = exports.ProductCreateRecurringPrices$inboundSchema;
    /** @deprecated use `ProductCreateRecurringPrices$outboundSchema` instead. */
    ProductCreateRecurringPrices$.outboundSchema = exports.ProductCreateRecurringPrices$outboundSchema;
})(ProductCreateRecurringPrices$ || (exports.ProductCreateRecurringPrices$ = ProductCreateRecurringPrices$ = {}));
function productCreateRecurringPricesToJSON(productCreateRecurringPrices) {
    return JSON.stringify(exports.ProductCreateRecurringPrices$outboundSchema.parse(productCreateRecurringPrices));
}
function productCreateRecurringPricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateRecurringPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurringPrices' from JSON`);
}
/** @internal */
exports.ProductCreateRecurring$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
    trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema).optional(),
    trial_interval_count: z.nullable(z.number().int()).optional(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductCreateRecurring$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
    trialInterval: z.nullable(trialinterval_js_1.TrialInterval$outboundSchema).optional(),
    trialIntervalCount: z.nullable(z.number().int()).optional(),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateRecurring$;
(function (ProductCreateRecurring$) {
    /** @deprecated use `ProductCreateRecurring$inboundSchema` instead. */
    ProductCreateRecurring$.inboundSchema = exports.ProductCreateRecurring$inboundSchema;
    /** @deprecated use `ProductCreateRecurring$outboundSchema` instead. */
    ProductCreateRecurring$.outboundSchema = exports.ProductCreateRecurring$outboundSchema;
})(ProductCreateRecurring$ || (exports.ProductCreateRecurring$ = ProductCreateRecurring$ = {}));
function productCreateRecurringToJSON(productCreateRecurring) {
    return JSON.stringify(exports.ProductCreateRecurring$outboundSchema.parse(productCreateRecurring));
}
function productCreateRecurringFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateRecurring$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurring' from JSON`);
}
