// @polar-sh/sdk/dist/commonjs/models/components/productupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductUpdate$ = exports.ProductUpdate$outboundSchema = exports.ProductUpdate$inboundSchema = exports.ProductUpdatePrices$ = exports.ProductUpdatePrices$outboundSchema = exports.ProductUpdatePrices$inboundSchema = exports.Two$ = exports.Two$outboundSchema = exports.Two$inboundSchema = exports.ProductUpdateMetadata$ = exports.ProductUpdateMetadata$outboundSchema = exports.ProductUpdateMetadata$inboundSchema = void 0;
exports.productUpdateMetadataToJSON = productUpdateMetadataToJSON;
exports.productUpdateMetadataFromJSON = productUpdateMetadataFromJSON;
exports.twoToJSON = twoToJSON;
exports.twoFromJSON = twoFromJSON;
exports.productUpdatePricesToJSON = productUpdatePricesToJSON;
exports.productUpdatePricesFromJSON = productUpdatePricesFromJSON;
exports.productUpdateToJSON = productUpdateToJSON;
exports.productUpdateFromJSON = productUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const existingproductprice_js_1 = require("./existingproductprice.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const productpriceseatbasedcreate_js_1 = require("./productpriceseatbasedcreate.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.ProductUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.ProductUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductUpdateMetadata$;
(function (ProductUpdateMetadata$) {
    /** @deprecated use `ProductUpdateMetadata$inboundSchema` instead. */
    ProductUpdateMetadata$.inboundSchema = exports.ProductUpdateMetadata$inboundSchema;
    /** @deprecated use `ProductUpdateMetadata$outboundSchema` instead. */
    ProductUpdateMetadata$.outboundSchema = exports.ProductUpdateMetadata$outboundSchema;
})(ProductUpdateMetadata$ || (exports.ProductUpdateMetadata$ = ProductUpdateMetadata$ = {}));
function productUpdateMetadataToJSON(productUpdateMetadata) {
    return JSON.stringify(exports.ProductUpdateMetadata$outboundSchema.parse(productUpdateMetadata));
}
function productUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdateMetadata' from JSON`);
}
/** @internal */
exports.Two$inboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
exports.Two$outboundSchema = z
    .union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Two$;
(function (Two$) {
    /** @deprecated use `Two$inboundSchema` instead. */
    Two$.inboundSchema = exports.Two$inboundSchema;
    /** @deprecated use `Two$outboundSchema` instead. */
    Two$.outboundSchema = exports.Two$outboundSchema;
})(Two$ || (exports.Two$ = Two$ = {}));
function twoToJSON(two) {
    return JSON.stringify(exports.Two$outboundSchema.parse(two));
}
function twoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Two$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Two' from JSON`);
}
/** @internal */
exports.ProductUpdatePrices$inboundSchema = z.union([
    existingproductprice_js_1.ExistingProductPrice$inboundSchema,
    z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ]),
]);
/** @internal */
exports.ProductUpdatePrices$outboundSchema = z.union([
    existingproductprice_js_1.ExistingProductPrice$outboundSchema,
    z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ]),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductUpdatePrices$;
(function (ProductUpdatePrices$) {
    /** @deprecated use `ProductUpdatePrices$inboundSchema` instead. */
    ProductUpdatePrices$.inboundSchema = exports.ProductUpdatePrices$inboundSchema;
    /** @deprecated use `ProductUpdatePrices$outboundSchema` instead. */
    ProductUpdatePrices$.outboundSchema = exports.ProductUpdatePrices$outboundSchema;
})(ProductUpdatePrices$ || (exports.ProductUpdatePrices$ = ProductUpdatePrices$ = {}));
function productUpdatePricesToJSON(productUpdatePrices) {
    return JSON.stringify(exports.ProductUpdatePrices$outboundSchema.parse(productUpdatePrices));
}
function productUpdatePricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductUpdatePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdatePrices' from JSON`);
}
/** @internal */
exports.ProductUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema).optional(),
    trial_interval_count: z.nullable(z.number().int()).optional(),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema)
        .optional(),
    is_archived: z.nullable(z.boolean()).optional(),
    prices: z.nullable(z.array(z.union([
        existingproductprice_js_1.ExistingProductPrice$inboundSchema,
        z.union([
            productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({ amountType: v.amount_type }))),
            productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
            productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({ amountType: v.amount_type }))),
            productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
            productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
        ]),
    ]))).optional(),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.nullable(z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "is_archived": "isArchived",
        "attached_custom_fields": "attachedCustomFields",
    });
});
/** @internal */
exports.ProductUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trialInterval: z.nullable(trialinterval_js_1.TrialInterval$outboundSchema).optional(),
    trialIntervalCount: z.nullable(z.number().int()).optional(),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema)
        .optional(),
    isArchived: z.nullable(z.boolean()).optional(),
    prices: z.nullable(z.array(z.union([
        existingproductprice_js_1.ExistingProductPrice$outboundSchema,
        z.union([
            productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({ amount_type: v.amountType }))),
            productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
            productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({ amount_type: v.amountType }))),
            productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
            productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
        ]),
    ]))).optional(),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.nullable(z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        isArchived: "is_archived",
        attachedCustomFields: "attached_custom_fields",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductUpdate$;
(function (ProductUpdate$) {
    /** @deprecated use `ProductUpdate$inboundSchema` instead. */
    ProductUpdate$.inboundSchema = exports.ProductUpdate$inboundSchema;
    /** @deprecated use `ProductUpdate$outboundSchema` instead. */
    ProductUpdate$.outboundSchema = exports.ProductUpdate$outboundSchema;
})(ProductUpdate$ || (exports.ProductUpdate$ = ProductUpdate$ = {}));
function productUpdateToJSON(productUpdate) {
    return JSON.stringify(exports.ProductUpdate$outboundSchema.parse(productUpdate));
}
function productUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdate' from JSON`);
}
