// @polar-sh/sdk/dist/commonjs/models/components/productpriceseatbasedcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceSeatBasedCreate$ = exports.ProductPriceSeatBasedCreate$outboundSchema = exports.ProductPriceSeatBasedCreate$inboundSchema = void 0;
exports.productPriceSeatBasedCreateToJSON = productPriceSeatBasedCreateToJSON;
exports.productPriceSeatBasedCreateFromJSON = productPriceSeatBasedCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const productpriceseattiers_js_1 = require("./productpriceseattiers.js");
/** @internal */
exports.ProductPriceSeatBasedCreate$inboundSchema = z.object({
    amount_type: z.literal("seat_based"),
    price_currency: z.string().default("usd"),
    seat_tiers: productpriceseattiers_js_1.ProductPriceSeatTiers$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "seat_tiers": "seatTiers",
    });
});
/** @internal */
exports.ProductPriceSeatBasedCreate$outboundSchema = z.object({
    amountType: z.literal("seat_based"),
    priceCurrency: z.string().default("usd"),
    seatTiers: productpriceseattiers_js_1.ProductPriceSeatTiers$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        seatTiers: "seat_tiers",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceSeatBasedCreate$;
(function (ProductPriceSeatBasedCreate$) {
    /** @deprecated use `ProductPriceSeatBasedCreate$inboundSchema` instead. */
    ProductPriceSeatBasedCreate$.inboundSchema = exports.ProductPriceSeatBasedCreate$inboundSchema;
    /** @deprecated use `ProductPriceSeatBasedCreate$outboundSchema` instead. */
    ProductPriceSeatBasedCreate$.outboundSchema = exports.ProductPriceSeatBasedCreate$outboundSchema;
})(ProductPriceSeatBasedCreate$ || (exports.ProductPriceSeatBasedCreate$ = ProductPriceSeatBasedCreate$ = {}));
function productPriceSeatBasedCreateToJSON(productPriceSeatBasedCreate) {
    return JSON.stringify(exports.ProductPriceSeatBasedCreate$outboundSchema.parse(productPriceSeatBasedCreate));
}
function productPriceSeatBasedCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceSeatBasedCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatBasedCreate' from JSON`);
}
