// @polar-sh/sdk/dist/commonjs/models/components/productcreateonetime.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCreateOneTime$ = exports.ProductCreateOneTime$outboundSchema = exports.ProductCreateOneTime$inboundSchema = exports.ProductCreateOneTimePrices$ = exports.ProductCreateOneTimePrices$outboundSchema = exports.ProductCreateOneTimePrices$inboundSchema = exports.ProductCreateOneTimeMetadata$ = exports.ProductCreateOneTimeMetadata$outboundSchema = exports.ProductCreateOneTimeMetadata$inboundSchema = void 0;
exports.productCreateOneTimeMetadataToJSON = productCreateOneTimeMetadataToJSON;
exports.productCreateOneTimeMetadataFromJSON = productCreateOneTimeMetadataFromJSON;
exports.productCreateOneTimePricesToJSON = productCreateOneTimePricesToJSON;
exports.productCreateOneTimePricesFromJSON = productCreateOneTimePricesFromJSON;
exports.productCreateOneTimeToJSON = productCreateOneTimeToJSON;
exports.productCreateOneTimeFromJSON = productCreateOneTimeFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const productpriceseatbasedcreate_js_1 = require("./productpriceseatbasedcreate.js");
/** @internal */
exports.ProductCreateOneTimeMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.ProductCreateOneTimeMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateOneTimeMetadata$;
(function (ProductCreateOneTimeMetadata$) {
    /** @deprecated use `ProductCreateOneTimeMetadata$inboundSchema` instead. */
    ProductCreateOneTimeMetadata$.inboundSchema = exports.ProductCreateOneTimeMetadata$inboundSchema;
    /** @deprecated use `ProductCreateOneTimeMetadata$outboundSchema` instead. */
    ProductCreateOneTimeMetadata$.outboundSchema = exports.ProductCreateOneTimeMetadata$outboundSchema;
})(ProductCreateOneTimeMetadata$ || (exports.ProductCreateOneTimeMetadata$ = ProductCreateOneTimeMetadata$ = {}));
function productCreateOneTimeMetadataToJSON(productCreateOneTimeMetadata) {
    return JSON.stringify(exports.ProductCreateOneTimeMetadata$outboundSchema.parse(productCreateOneTimeMetadata));
}
function productCreateOneTimeMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateOneTimeMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTimeMetadata' from JSON`);
}
/** @internal */
exports.ProductCreateOneTimePrices$inboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
exports.ProductCreateOneTimePrices$outboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateOneTimePrices$;
(function (ProductCreateOneTimePrices$) {
    /** @deprecated use `ProductCreateOneTimePrices$inboundSchema` instead. */
    ProductCreateOneTimePrices$.inboundSchema = exports.ProductCreateOneTimePrices$inboundSchema;
    /** @deprecated use `ProductCreateOneTimePrices$outboundSchema` instead. */
    ProductCreateOneTimePrices$.outboundSchema = exports.ProductCreateOneTimePrices$outboundSchema;
})(ProductCreateOneTimePrices$ || (exports.ProductCreateOneTimePrices$ = ProductCreateOneTimePrices$ = {}));
function productCreateOneTimePricesToJSON(productCreateOneTimePrices) {
    return JSON.stringify(exports.ProductCreateOneTimePrices$outboundSchema.parse(productCreateOneTimePrices));
}
function productCreateOneTimePricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateOneTimePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTimePrices' from JSON`);
}
/** @internal */
exports.ProductCreateOneTime$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductCreateOneTime$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateOneTime$;
(function (ProductCreateOneTime$) {
    /** @deprecated use `ProductCreateOneTime$inboundSchema` instead. */
    ProductCreateOneTime$.inboundSchema = exports.ProductCreateOneTime$inboundSchema;
    /** @deprecated use `ProductCreateOneTime$outboundSchema` instead. */
    ProductCreateOneTime$.outboundSchema = exports.ProductCreateOneTime$outboundSchema;
})(ProductCreateOneTime$ || (exports.ProductCreateOneTime$ = ProductCreateOneTime$ = {}));
function productCreateOneTimeToJSON(productCreateOneTime) {
    return JSON.stringify(exports.ProductCreateOneTime$outboundSchema.parse(productCreateOneTime));
}
function productCreateOneTimeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateOneTime$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTime' from JSON`);
}
