// @polar-sh/sdk/dist/commonjs/models/components/webtokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTokenRequest$ = exports.WebTokenRequest$outboundSchema = exports.WebTokenRequest$inboundSchema = exports.WebTokenRequestSubType$ = exports.WebTokenRequestSubType$outboundSchema = exports.WebTokenRequestSubType$inboundSchema = exports.WebTokenRequestSubType = void 0;
exports.webTokenRequestToJSON = webTokenRequestToJSON;
exports.webTokenRequestFromJSON = webTokenRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.WebTokenRequestSubType = {
    User: "user",
    Organization: "organization",
};
/** @internal */
exports.WebTokenRequestSubType$inboundSchema = z.nativeEnum(exports.WebTokenRequestSubType);
/** @internal */
exports.WebTokenRequestSubType$outboundSchema = exports.WebTokenRequestSubType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebTokenRequestSubType$;
(function (WebTokenRequestSubType$) {
    /** @deprecated use `WebTokenRequestSubType$inboundSchema` instead. */
    WebTokenRequestSubType$.inboundSchema = exports.WebTokenRequestSubType$inboundSchema;
    /** @deprecated use `WebTokenRequestSubType$outboundSchema` instead. */
    WebTokenRequestSubType$.outboundSchema = exports.WebTokenRequestSubType$outboundSchema;
})(WebTokenRequestSubType$ || (exports.WebTokenRequestSubType$ = WebTokenRequestSubType$ = {}));
/** @internal */
exports.WebTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("web"),
    client_id: z.string(),
    client_secret: z.string(),
    session_token: z.string(),
    sub_type: exports.WebTokenRequestSubType$inboundSchema.default("user"),
    sub: z.nullable(z.string()).optional(),
    scope: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "session_token": "sessionToken",
        "sub_type": "subType",
    });
});
/** @internal */
exports.WebTokenRequest$outboundSchema = z.object({
    grantType: z.literal("web"),
    clientId: z.string(),
    clientSecret: z.string(),
    sessionToken: z.string(),
    subType: exports.WebTokenRequestSubType$outboundSchema.default("user"),
    sub: z.nullable(z.string()).optional(),
    scope: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        sessionToken: "session_token",
        subType: "sub_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebTokenRequest$;
(function (WebTokenRequest$) {
    /** @deprecated use `WebTokenRequest$inboundSchema` instead. */
    WebTokenRequest$.inboundSchema = exports.WebTokenRequest$inboundSchema;
    /** @deprecated use `WebTokenRequest$outboundSchema` instead. */
    WebTokenRequest$.outboundSchema = exports.WebTokenRequest$outboundSchema;
})(WebTokenRequest$ || (exports.WebTokenRequest$ = WebTokenRequest$ = {}));
function webTokenRequestToJSON(webTokenRequest) {
    return JSON.stringify(exports.WebTokenRequest$outboundSchema.parse(webTokenRequest));
}
function webTokenRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebTokenRequest' from JSON`);
}
