// @polar-sh/sdk/dist/esm/models/components/aggregationfunction.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const AggregationFunction = {
    Count: "count",
    Sum: "sum",
    Max: "max",
    Min: "min",
    Avg: "avg",
    Unique: "unique",
};
/** @internal */
export const AggregationFunction$inboundSchema = z.nativeEnum(AggregationFunction);
/** @internal */
export const AggregationFunction$outboundSchema = AggregationFunction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var AggregationFunction$;
(function (AggregationFunction$) {
    /** @deprecated use `AggregationFunction$inboundSchema` instead. */
    AggregationFunction$.inboundSchema = AggregationFunction$inboundSchema;
    /** @deprecated use `AggregationFunction$outboundSchema` instead. */
    AggregationFunction$.outboundSchema = AggregationFunction$outboundSchema;
})(AggregationFunction$ || (AggregationFunction$ = {}));
