// @polar-sh/sdk/dist/esm/models/components/benefitgrantsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const BenefitGrantSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    GrantedAt: "granted_at",
    MinusGrantedAt: "-granted_at",
    RevokedAt: "revoked_at",
    MinusRevokedAt: "-revoked_at",
};
/** @internal */
export const BenefitGrantSortProperty$inboundSchema = z.nativeEnum(BenefitGrantSortProperty);
/** @internal */
export const BenefitGrantSortProperty$outboundSchema = BenefitGrantSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantSortProperty$;
(function (BenefitGrantSortProperty$) {
    /** @deprecated use `BenefitGrantSortProperty$inboundSchema` instead. */
    BenefitGrantSortProperty$.inboundSchema = BenefitGrantSortProperty$inboundSchema;
    /** @deprecated use `BenefitGrantSortProperty$outboundSchema` instead. */
    BenefitGrantSortProperty$.outboundSchema = BenefitGrantSortProperty$outboundSchema;
})(BenefitGrantSortProperty$ || (BenefitGrantSortProperty$ = {}));
