// @polar-sh/sdk/dist/esm/models/components/checkoutlinkcreateproductprice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CheckoutLinkCreateProductPriceMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CheckoutLinkCreateProductPriceMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProductPriceMetadata$;
(function (CheckoutLinkCreateProductPriceMetadata$) {
    /** @deprecated use `CheckoutLinkCreateProductPriceMetadata$inboundSchema` instead. */
    CheckoutLinkCreateProductPriceMetadata$.inboundSchema = CheckoutLinkCreateProductPriceMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductPriceMetadata$outboundSchema` instead. */
    CheckoutLinkCreateProductPriceMetadata$.outboundSchema = CheckoutLinkCreateProductPriceMetadata$outboundSchema;
})(CheckoutLinkCreateProductPriceMetadata$ || (CheckoutLinkCreateProductPriceMetadata$ = {}));
export function checkoutLinkCreateProductPriceMetadataToJSON(checkoutLinkCreateProductPriceMetadata) {
    return JSON.stringify(CheckoutLinkCreateProductPriceMetadata$outboundSchema.parse(checkoutLinkCreateProductPriceMetadata));
}
export function checkoutLinkCreateProductPriceMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProductPriceMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductPriceMetadata' from JSON`);
}
/** @internal */
export const CheckoutLinkCreateProductPrice$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trial_interval: z.nullable(TrialInterval$inboundSchema).optional(),
    trial_interval_count: z.nullable(z.number().int()).optional(),
    payment_processor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    require_billing_address: z.boolean().default(false),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    product_price_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
        "product_price_id": "productPriceId",
    });
});
/** @internal */
export const CheckoutLinkCreateProductPrice$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trialInterval: z.nullable(TrialInterval$outboundSchema).optional(),
    trialIntervalCount: z.nullable(z.number().int()).optional(),
    paymentProcessor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    requireBillingAddress: z.boolean().default(false),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    productPriceId: z.string(),
}).transform((v) => {
    return remap$(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
        productPriceId: "product_price_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProductPrice$;
(function (CheckoutLinkCreateProductPrice$) {
    /** @deprecated use `CheckoutLinkCreateProductPrice$inboundSchema` instead. */
    CheckoutLinkCreateProductPrice$.inboundSchema = CheckoutLinkCreateProductPrice$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductPrice$outboundSchema` instead. */
    CheckoutLinkCreateProductPrice$.outboundSchema = CheckoutLinkCreateProductPrice$outboundSchema;
})(CheckoutLinkCreateProductPrice$ || (CheckoutLinkCreateProductPrice$ = {}));
export function checkoutLinkCreateProductPriceToJSON(checkoutLinkCreateProductPrice) {
    return JSON.stringify(CheckoutLinkCreateProductPrice$outboundSchema.parse(checkoutLinkCreateProductPrice));
}
export function checkoutLinkCreateProductPriceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductPrice' from JSON`);
}
