// @polar-sh/sdk/dist/esm/models/components/customerpaymentmethodcreateresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema, CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema, } from "./customerpaymentmethodcreaterequiresactionresponse.js";
import { CustomerPaymentMethodCreateSucceededResponse$inboundSchema, CustomerPaymentMethodCreateSucceededResponse$outboundSchema, } from "./customerpaymentmethodcreatesucceededresponse.js";
/** @internal */
export const CustomerPaymentMethodCreateResponse$inboundSchema = z.union([
    CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema.and(z.object({ status: z.literal("requires_action") }).transform((v) => ({
        status: v.status,
    }))),
    CustomerPaymentMethodCreateSucceededResponse$inboundSchema.and(z.object({ status: z.literal("succeeded") }).transform((v) => ({
        status: v.status,
    }))),
]);
/** @internal */
export const CustomerPaymentMethodCreateResponse$outboundSchema = z.union([
    CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema.and(z.object({ status: z.literal("requires_action") }).transform((v) => ({
        status: v.status,
    }))),
    CustomerPaymentMethodCreateSucceededResponse$outboundSchema.and(z.object({ status: z.literal("succeeded") }).transform((v) => ({
        status: v.status,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPaymentMethodCreateResponse$;
(function (CustomerPaymentMethodCreateResponse$) {
    /** @deprecated use `CustomerPaymentMethodCreateResponse$inboundSchema` instead. */
    CustomerPaymentMethodCreateResponse$.inboundSchema = CustomerPaymentMethodCreateResponse$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodCreateResponse$outboundSchema` instead. */
    CustomerPaymentMethodCreateResponse$.outboundSchema = CustomerPaymentMethodCreateResponse$outboundSchema;
})(CustomerPaymentMethodCreateResponse$ || (CustomerPaymentMethodCreateResponse$ = {}));
export function customerPaymentMethodCreateResponseToJSON(customerPaymentMethodCreateResponse) {
    return JSON.stringify(CustomerPaymentMethodCreateResponse$outboundSchema.parse(customerPaymentMethodCreateResponse));
}
export function customerPaymentMethodCreateResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPaymentMethodCreateResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodCreateResponse' from JSON`);
}
