// @polar-sh/sdk/dist/esm/models/components/customercustomersession.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerCustomerSession$inboundSchema = z.object({
    expires_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    return_url: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "expires_at": "expiresAt",
        "return_url": "returnUrl",
    });
});
/** @internal */
export const CustomerCustomerSession$outboundSchema = z.object({
    expiresAt: z.date().transform(v => v.toISOString()),
    returnUrl: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        expiresAt: "expires_at",
        returnUrl: "return_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerCustomerSession$;
(function (CustomerCustomerSession$) {
    /** @deprecated use `CustomerCustomerSession$inboundSchema` instead. */
    CustomerCustomerSession$.inboundSchema = CustomerCustomerSession$inboundSchema;
    /** @deprecated use `CustomerCustomerSession$outboundSchema` instead. */
    CustomerCustomerSession$.outboundSchema = CustomerCustomerSession$outboundSchema;
})(CustomerCustomerSession$ || (CustomerCustomerSession$ = {}));
export function customerCustomerSessionToJSON(customerCustomerSession) {
    return JSON.stringify(CustomerCustomerSession$outboundSchema.parse(customerCustomerSession));
}
export function customerCustomerSessionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerCustomerSession$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerCustomerSession' from JSON`);
}
