// @polar-sh/sdk/dist/esm/models/components/customerproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitPublic$inboundSchema, BenefitPublic$outboundSchema, } from "./benefitpublic.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { ProductMediaFileRead$inboundSchema, ProductMediaFileRead$outboundSchema, } from "./productmediafileread.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CustomerProductPrices$inboundSchema = z.union([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CustomerProductPrices$outboundSchema = z.union([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerProductPrices$;
(function (CustomerProductPrices$) {
    /** @deprecated use `CustomerProductPrices$inboundSchema` instead. */
    CustomerProductPrices$.inboundSchema = CustomerProductPrices$inboundSchema;
    /** @deprecated use `CustomerProductPrices$outboundSchema` instead. */
    CustomerProductPrices$.outboundSchema = CustomerProductPrices$outboundSchema;
})(CustomerProductPrices$ || (CustomerProductPrices$ = {}));
export function customerProductPricesToJSON(customerProductPrices) {
    return JSON.stringify(CustomerProductPrices$outboundSchema.parse(customerProductPrices));
}
export function customerProductPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerProductPrices' from JSON`);
}
/** @internal */
export const CustomerProduct$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    benefits: z.array(BenefitPublic$inboundSchema),
    medias: z.array(ProductMediaFileRead$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CustomerProduct$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    trialInterval: z.nullable(TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])),
    benefits: z.array(BenefitPublic$outboundSchema),
    medias: z.array(ProductMediaFileRead$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerProduct$;
(function (CustomerProduct$) {
    /** @deprecated use `CustomerProduct$inboundSchema` instead. */
    CustomerProduct$.inboundSchema = CustomerProduct$inboundSchema;
    /** @deprecated use `CustomerProduct$outboundSchema` instead. */
    CustomerProduct$.outboundSchema = CustomerProduct$outboundSchema;
})(CustomerProduct$ || (CustomerProduct$ = {}));
export function customerProductToJSON(customerProduct) {
    return JSON.stringify(CustomerProduct$outboundSchema.parse(customerProduct));
}
export function customerProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerProduct' from JSON`);
}
