// @polar-sh/sdk/dist/esm/models/components/customerpaymentmethodcreatesucceededresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerPaymentMethod$inboundSchema, CustomerPaymentMethod$outboundSchema, } from "./customerpaymentmethod.js";
/** @internal */
export const CustomerPaymentMethodCreateSucceededResponse$inboundSchema = z.object({
    status: z.literal("succeeded"),
    payment_method: CustomerPaymentMethod$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "payment_method": "paymentMethod",
    });
});
/** @internal */
export const CustomerPaymentMethodCreateSucceededResponse$outboundSchema = z.object({
    status: z.literal("succeeded"),
    paymentMethod: CustomerPaymentMethod$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        paymentMethod: "payment_method",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPaymentMethodCreateSucceededResponse$;
(function (CustomerPaymentMethodCreateSucceededResponse$) {
    /** @deprecated use `CustomerPaymentMethodCreateSucceededResponse$inboundSchema` instead. */
    CustomerPaymentMethodCreateSucceededResponse$.inboundSchema = CustomerPaymentMethodCreateSucceededResponse$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodCreateSucceededResponse$outboundSchema` instead. */
    CustomerPaymentMethodCreateSucceededResponse$.outboundSchema = CustomerPaymentMethodCreateSucceededResponse$outboundSchema;
})(CustomerPaymentMethodCreateSucceededResponse$ || (CustomerPaymentMethodCreateSucceededResponse$ = {}));
export function customerPaymentMethodCreateSucceededResponseToJSON(customerPaymentMethodCreateSucceededResponse) {
    return JSON.stringify(CustomerPaymentMethodCreateSucceededResponse$outboundSchema.parse(customerPaymentMethodCreateSucceededResponse));
}
export function customerPaymentMethodCreateSucceededResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPaymentMethodCreateSucceededResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodCreateSucceededResponse' from JSON`);
}
