// @polar-sh/sdk/dist/esm/models/components/customerseat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { SeatStatus$inboundSchema, SeatStatus$outboundSchema, } from "./seatstatus.js";
/** @internal */
export const CustomerSeat$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    subscription_id: z.string(),
    status: SeatStatus$inboundSchema,
    customer_id: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    invitation_token_expires_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    claimed_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    seat_metadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subscription_id": "subscriptionId",
        "customer_id": "customerId",
        "customer_email": "customerEmail",
        "invitation_token_expires_at": "invitationTokenExpiresAt",
        "claimed_at": "claimedAt",
        "revoked_at": "revokedAt",
        "seat_metadata": "seatMetadata",
    });
});
/** @internal */
export const CustomerSeat$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    subscriptionId: z.string(),
    status: SeatStatus$outboundSchema,
    customerId: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    invitationTokenExpiresAt: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
    claimedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    seatMetadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subscriptionId: "subscription_id",
        customerId: "customer_id",
        customerEmail: "customer_email",
        invitationTokenExpiresAt: "invitation_token_expires_at",
        claimedAt: "claimed_at",
        revokedAt: "revoked_at",
        seatMetadata: "seat_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeat$;
(function (CustomerSeat$) {
    /** @deprecated use `CustomerSeat$inboundSchema` instead. */
    CustomerSeat$.inboundSchema = CustomerSeat$inboundSchema;
    /** @deprecated use `CustomerSeat$outboundSchema` instead. */
    CustomerSeat$.outboundSchema = CustomerSeat$outboundSchema;
})(CustomerSeat$ || (CustomerSeat$ = {}));
export function customerSeatToJSON(customerSeat) {
    return JSON.stringify(CustomerSeat$outboundSchema.parse(customerSeat));
}
export function customerSeatFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeat$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeat' from JSON`);
}
