// @polar-sh/sdk/dist/esm/models/components/customerpaymentmethodconfirm.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPaymentMethodConfirm$inboundSchema = z.object({
    setup_intent_id: z.string(),
    set_default: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "setup_intent_id": "setupIntentId",
        "set_default": "setDefault",
    });
});
/** @internal */
export const CustomerPaymentMethodConfirm$outboundSchema = z.object({
    setupIntentId: z.string(),
    setDefault: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        setupIntentId: "setup_intent_id",
        setDefault: "set_default",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPaymentMethodConfirm$;
(function (CustomerPaymentMethodConfirm$) {
    /** @deprecated use `CustomerPaymentMethodConfirm$inboundSchema` instead. */
    CustomerPaymentMethodConfirm$.inboundSchema = CustomerPaymentMethodConfirm$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodConfirm$outboundSchema` instead. */
    CustomerPaymentMethodConfirm$.outboundSchema = CustomerPaymentMethodConfirm$outboundSchema;
})(CustomerPaymentMethodConfirm$ || (CustomerPaymentMethodConfirm$ = {}));
export function customerPaymentMethodConfirmToJSON(customerPaymentMethodConfirm) {
    return JSON.stringify(CustomerPaymentMethodConfirm$outboundSchema.parse(customerPaymentMethodConfirm));
}
export function customerPaymentMethodConfirmFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPaymentMethodConfirm$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodConfirm' from JSON`);
}
