// @polar-sh/sdk/dist/esm/models/components/customerbalance.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerBalance$inboundSchema = z.object({
    balance: z.number().int(),
    currency: z.string(),
});
/** @internal */
export const CustomerBalance$outboundSchema = z.object({
    balance: z.number().int(),
    currency: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBalance$;
(function (CustomerBalance$) {
    /** @deprecated use `CustomerBalance$inboundSchema` instead. */
    CustomerBalance$.inboundSchema = CustomerBalance$inboundSchema;
    /** @deprecated use `CustomerBalance$outboundSchema` instead. */
    CustomerBalance$.outboundSchema = CustomerBalance$outboundSchema;
})(CustomerBalance$ || (CustomerBalance$ = {}));
export function customerBalanceToJSON(customerBalance) {
    return JSON.stringify(CustomerBalance$outboundSchema.parse(customerBalance));
}
export function customerBalanceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBalance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBalance' from JSON`);
}
