// @polar-sh/sdk/dist/esm/models/components/customerorderproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitPublic$inboundSchema, BenefitPublic$outboundSchema, } from "./benefitpublic.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { Organization$inboundSchema, Organization$outboundSchema, } from "./organization.js";
import { ProductMediaFileRead$inboundSchema, ProductMediaFileRead$outboundSchema, } from "./productmediafileread.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CustomerOrderProductPrices$inboundSchema = z.union([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CustomerOrderProductPrices$outboundSchema = z.union([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerOrderProductPrices$;
(function (CustomerOrderProductPrices$) {
    /** @deprecated use `CustomerOrderProductPrices$inboundSchema` instead. */
    CustomerOrderProductPrices$.inboundSchema = CustomerOrderProductPrices$inboundSchema;
    /** @deprecated use `CustomerOrderProductPrices$outboundSchema` instead. */
    CustomerOrderProductPrices$.outboundSchema = CustomerOrderProductPrices$outboundSchema;
})(CustomerOrderProductPrices$ || (CustomerOrderProductPrices$ = {}));
export function customerOrderProductPricesToJSON(customerOrderProductPrices) {
    return JSON.stringify(CustomerOrderProductPrices$outboundSchema.parse(customerOrderProductPrices));
}
export function customerOrderProductPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrderProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderProductPrices' from JSON`);
}
/** @internal */
export const CustomerOrderProduct$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    benefits: z.array(BenefitPublic$inboundSchema),
    medias: z.array(ProductMediaFileRead$inboundSchema),
    organization: Organization$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CustomerOrderProduct$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    trialInterval: z.nullable(TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])),
    benefits: z.array(BenefitPublic$outboundSchema),
    medias: z.array(ProductMediaFileRead$outboundSchema),
    organization: Organization$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerOrderProduct$;
(function (CustomerOrderProduct$) {
    /** @deprecated use `CustomerOrderProduct$inboundSchema` instead. */
    CustomerOrderProduct$.inboundSchema = CustomerOrderProduct$inboundSchema;
    /** @deprecated use `CustomerOrderProduct$outboundSchema` instead. */
    CustomerOrderProduct$.outboundSchema = CustomerOrderProduct$outboundSchema;
})(CustomerOrderProduct$ || (CustomerOrderProduct$ = {}));
export function customerOrderProductToJSON(customerOrderProduct) {
    return JSON.stringify(CustomerOrderProduct$outboundSchema.parse(customerOrderProduct));
}
export function customerOrderProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrderProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderProduct' from JSON`);
}
