// @polar-sh/sdk/dist/esm/models/components/customerpaymentmethodcreaterequiresactionresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema = z.object({
    status: z.literal("requires_action"),
    client_secret: z.string(),
}).transform((v) => {
    return remap$(v, {
        "client_secret": "clientSecret",
    });
});
/** @internal */
export const CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema = z.object({
    status: z.literal("requires_action"),
    clientSecret: z.string(),
}).transform((v) => {
    return remap$(v, {
        clientSecret: "client_secret",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPaymentMethodCreateRequiresActionResponse$;
(function (CustomerPaymentMethodCreateRequiresActionResponse$) {
    /** @deprecated use `CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema` instead. */
    CustomerPaymentMethodCreateRequiresActionResponse$.inboundSchema = CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema` instead. */
    CustomerPaymentMethodCreateRequiresActionResponse$.outboundSchema = CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema;
})(CustomerPaymentMethodCreateRequiresActionResponse$ || (CustomerPaymentMethodCreateRequiresActionResponse$ = {}));
export function customerPaymentMethodCreateRequiresActionResponseToJSON(customerPaymentMethodCreateRequiresActionResponse) {
    return JSON.stringify(CustomerPaymentMethodCreateRequiresActionResponse$outboundSchema.parse(customerPaymentMethodCreateRequiresActionResponse));
}
export function customerPaymentMethodCreateRequiresActionResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPaymentMethodCreateRequiresActionResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodCreateRequiresActionResponse' from JSON`);
}
