// @polar-sh/sdk/dist/esm/models/components/oauth2clientconfiguration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
export const TokenEndpointAuthMethod = {
    ClientSecretBasic: "client_secret_basic",
    ClientSecretPost: "client_secret_post",
    None: "none",
};
export const GrantTypes = {
    AuthorizationCode: "authorization_code",
    RefreshToken: "refresh_token",
};
/** @internal */
export const TokenEndpointAuthMethod$inboundSchema = z.nativeEnum(TokenEndpointAuthMethod);
/** @internal */
export const TokenEndpointAuthMethod$outboundSchema = TokenEndpointAuthMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TokenEndpointAuthMethod$;
(function (TokenEndpointAuthMethod$) {
    /** @deprecated use `TokenEndpointAuthMethod$inboundSchema` instead. */
    TokenEndpointAuthMethod$.inboundSchema = TokenEndpointAuthMethod$inboundSchema;
    /** @deprecated use `TokenEndpointAuthMethod$outboundSchema` instead. */
    TokenEndpointAuthMethod$.outboundSchema = TokenEndpointAuthMethod$outboundSchema;
})(TokenEndpointAuthMethod$ || (TokenEndpointAuthMethod$ = {}));
/** @internal */
export const GrantTypes$inboundSchema = z
    .nativeEnum(GrantTypes);
/** @internal */
export const GrantTypes$outboundSchema = GrantTypes$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GrantTypes$;
(function (GrantTypes$) {
    /** @deprecated use `GrantTypes$inboundSchema` instead. */
    GrantTypes$.inboundSchema = GrantTypes$inboundSchema;
    /** @deprecated use `GrantTypes$outboundSchema` instead. */
    GrantTypes$.outboundSchema = GrantTypes$outboundSchema;
})(GrantTypes$ || (GrantTypes$ = {}));
/** @internal */
export const OAuth2ClientConfiguration$inboundSchema = z.object({
    redirect_uris: z.array(z.string()),
    token_endpoint_auth_method: TokenEndpointAuthMethod$inboundSchema.default("client_secret_post"),
    grant_types: z.array(GrantTypes$inboundSchema).optional(),
    response_types: z.array(z.string()).optional(),
    scope: z.string().default("openid profile email user:read organizations:read organizations:write custom_fields:read custom_fields:write discounts:read discounts:write checkout_links:read checkout_links:write checkouts:read checkouts:write transactions:read transactions:write payouts:read payouts:write products:read products:write benefits:read benefits:write events:read events:write meters:read meters:write files:read files:write subscriptions:read subscriptions:write customers:read customers:write customer_meters:read customer_sessions:write orders:read orders:write refunds:read refunds:write payments:read metrics:read webhooks:read webhooks:write external_organizations:read license_keys:read license_keys:write repositories:read repositories:write issues:read issues:write customer_portal:read customer_portal:write notifications:read notifications:write notification_recipients:read notification_recipients:write"),
    client_name: z.string(),
    client_uri: z.nullable(z.string()).optional(),
    logo_uri: z.nullable(z.string()).optional(),
    tos_uri: z.nullable(z.string()).optional(),
    policy_uri: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "redirect_uris": "redirectUris",
        "token_endpoint_auth_method": "tokenEndpointAuthMethod",
        "grant_types": "grantTypes",
        "response_types": "responseTypes",
        "client_name": "clientName",
        "client_uri": "clientUri",
        "logo_uri": "logoUri",
        "tos_uri": "tosUri",
        "policy_uri": "policyUri",
    });
});
/** @internal */
export const OAuth2ClientConfiguration$outboundSchema = z.object({
    redirectUris: z.array(z.string()),
    tokenEndpointAuthMethod: TokenEndpointAuthMethod$outboundSchema.default("client_secret_post"),
    grantTypes: z.array(GrantTypes$outboundSchema).optional(),
    responseTypes: z.array(z.string()).optional(),
    scope: z.string().default("openid profile email user:read organizations:read organizations:write custom_fields:read custom_fields:write discounts:read discounts:write checkout_links:read checkout_links:write checkouts:read checkouts:write transactions:read transactions:write payouts:read payouts:write products:read products:write benefits:read benefits:write events:read events:write meters:read meters:write files:read files:write subscriptions:read subscriptions:write customers:read customers:write customer_meters:read customer_sessions:write orders:read orders:write refunds:read refunds:write payments:read metrics:read webhooks:read webhooks:write external_organizations:read license_keys:read license_keys:write repositories:read repositories:write issues:read issues:write customer_portal:read customer_portal:write notifications:read notifications:write notification_recipients:read notification_recipients:write"),
    clientName: z.string(),
    clientUri: z.nullable(z.string()).optional(),
    logoUri: z.nullable(z.string()).optional(),
    tosUri: z.nullable(z.string()).optional(),
    policyUri: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        redirectUris: "redirect_uris",
        tokenEndpointAuthMethod: "token_endpoint_auth_method",
        grantTypes: "grant_types",
        responseTypes: "response_types",
        clientName: "client_name",
        clientUri: "client_uri",
        logoUri: "logo_uri",
        tosUri: "tos_uri",
        policyUri: "policy_uri",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OAuth2ClientConfiguration$;
(function (OAuth2ClientConfiguration$) {
    /** @deprecated use `OAuth2ClientConfiguration$inboundSchema` instead. */
    OAuth2ClientConfiguration$.inboundSchema = OAuth2ClientConfiguration$inboundSchema;
    /** @deprecated use `OAuth2ClientConfiguration$outboundSchema` instead. */
    OAuth2ClientConfiguration$.outboundSchema = OAuth2ClientConfiguration$outboundSchema;
})(OAuth2ClientConfiguration$ || (OAuth2ClientConfiguration$ = {}));
export function oAuth2ClientConfigurationToJSON(oAuth2ClientConfiguration) {
    return JSON.stringify(OAuth2ClientConfiguration$outboundSchema.parse(oAuth2ClientConfiguration));
}
export function oAuth2ClientConfigurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OAuth2ClientConfiguration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OAuth2ClientConfiguration' from JSON`);
}
