// @polar-sh/sdk/dist/esm/models/components/orderproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const OrderProductMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const OrderProductMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderProductMetadata$;
(function (OrderProductMetadata$) {
    /** @deprecated use `OrderProductMetadata$inboundSchema` instead. */
    OrderProductMetadata$.inboundSchema = OrderProductMetadata$inboundSchema;
    /** @deprecated use `OrderProductMetadata$outboundSchema` instead. */
    OrderProductMetadata$.outboundSchema = OrderProductMetadata$outboundSchema;
})(OrderProductMetadata$ || (OrderProductMetadata$ = {}));
export function orderProductMetadataToJSON(orderProductMetadata) {
    return JSON.stringify(OrderProductMetadata$outboundSchema.parse(orderProductMetadata));
}
export function orderProductMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderProductMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderProductMetadata' from JSON`);
}
/** @internal */
export const OrderProduct$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const OrderProduct$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    trialInterval: z.nullable(TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.number().int()),
    name: z.string(),
    description: z.nullable(z.string()),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderProduct$;
(function (OrderProduct$) {
    /** @deprecated use `OrderProduct$inboundSchema` instead. */
    OrderProduct$.inboundSchema = OrderProduct$inboundSchema;
    /** @deprecated use `OrderProduct$outboundSchema` instead. */
    OrderProduct$.outboundSchema = OrderProduct$outboundSchema;
})(OrderProduct$ || (OrderProduct$ = {}));
export function orderProductToJSON(orderProduct) {
    return JSON.stringify(OrderProduct$outboundSchema.parse(orderProduct));
}
export function orderProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderProduct' from JSON`);
}
