// @polar-sh/sdk/dist/esm/models/components/productcreateonetime.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { AttachedCustomFieldCreate$inboundSchema, AttachedCustomFieldCreate$outboundSchema, } from "./attachedcustomfieldcreate.js";
import { ProductPriceCustomCreate$inboundSchema, ProductPriceCustomCreate$outboundSchema, } from "./productpricecustomcreate.js";
import { ProductPriceFixedCreate$inboundSchema, ProductPriceFixedCreate$outboundSchema, } from "./productpricefixedcreate.js";
import { ProductPriceFreeCreate$inboundSchema, ProductPriceFreeCreate$outboundSchema, } from "./productpricefreecreate.js";
import { ProductPriceMeteredUnitCreate$inboundSchema, ProductPriceMeteredUnitCreate$outboundSchema, } from "./productpricemeteredunitcreate.js";
import { ProductPriceSeatBasedCreate$inboundSchema, ProductPriceSeatBasedCreate$outboundSchema, } from "./productpriceseatbasedcreate.js";
/** @internal */
export const ProductCreateOneTimeMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const ProductCreateOneTimeMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateOneTimeMetadata$;
(function (ProductCreateOneTimeMetadata$) {
    /** @deprecated use `ProductCreateOneTimeMetadata$inboundSchema` instead. */
    ProductCreateOneTimeMetadata$.inboundSchema = ProductCreateOneTimeMetadata$inboundSchema;
    /** @deprecated use `ProductCreateOneTimeMetadata$outboundSchema` instead. */
    ProductCreateOneTimeMetadata$.outboundSchema = ProductCreateOneTimeMetadata$outboundSchema;
})(ProductCreateOneTimeMetadata$ || (ProductCreateOneTimeMetadata$ = {}));
export function productCreateOneTimeMetadataToJSON(productCreateOneTimeMetadata) {
    return JSON.stringify(ProductCreateOneTimeMetadata$outboundSchema.parse(productCreateOneTimeMetadata));
}
export function productCreateOneTimeMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateOneTimeMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTimeMetadata' from JSON`);
}
/** @internal */
export const ProductCreateOneTimePrices$inboundSchema = z.union([
    ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
export const ProductCreateOneTimePrices$outboundSchema = z.union([
    ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateOneTimePrices$;
(function (ProductCreateOneTimePrices$) {
    /** @deprecated use `ProductCreateOneTimePrices$inboundSchema` instead. */
    ProductCreateOneTimePrices$.inboundSchema = ProductCreateOneTimePrices$inboundSchema;
    /** @deprecated use `ProductCreateOneTimePrices$outboundSchema` instead. */
    ProductCreateOneTimePrices$.outboundSchema = ProductCreateOneTimePrices$outboundSchema;
})(ProductCreateOneTimePrices$ || (ProductCreateOneTimePrices$ = {}));
export function productCreateOneTimePricesToJSON(productCreateOneTimePrices) {
    return JSON.stringify(ProductCreateOneTimePrices$outboundSchema.parse(productCreateOneTimePrices));
}
export function productCreateOneTimePricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateOneTimePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTimePrices' from JSON`);
}
/** @internal */
export const ProductCreateOneTime$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(z.any()).optional(),
}).transform((v) => {
    return remap$(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
export const ProductCreateOneTime$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(z.any()).optional(),
}).transform((v) => {
    return remap$(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateOneTime$;
(function (ProductCreateOneTime$) {
    /** @deprecated use `ProductCreateOneTime$inboundSchema` instead. */
    ProductCreateOneTime$.inboundSchema = ProductCreateOneTime$inboundSchema;
    /** @deprecated use `ProductCreateOneTime$outboundSchema` instead. */
    ProductCreateOneTime$.outboundSchema = ProductCreateOneTime$outboundSchema;
})(ProductCreateOneTime$ || (ProductCreateOneTime$ = {}));
export function productCreateOneTimeToJSON(productCreateOneTime) {
    return JSON.stringify(ProductCreateOneTime$outboundSchema.parse(productCreateOneTime));
}
export function productCreateOneTimeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateOneTime$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateOneTime' from JSON`);
}
