// @polar-sh/sdk/dist/esm/models/components/productcreaterecurring.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { AttachedCustomFieldCreate$inboundSchema, AttachedCustomFieldCreate$outboundSchema, } from "./attachedcustomfieldcreate.js";
import { ProductPriceCustomCreate$inboundSchema, ProductPriceCustomCreate$outboundSchema, } from "./productpricecustomcreate.js";
import { ProductPriceFixedCreate$inboundSchema, ProductPriceFixedCreate$outboundSchema, } from "./productpricefixedcreate.js";
import { ProductPriceFreeCreate$inboundSchema, ProductPriceFreeCreate$outboundSchema, } from "./productpricefreecreate.js";
import { ProductPriceMeteredUnitCreate$inboundSchema, ProductPriceMeteredUnitCreate$outboundSchema, } from "./productpricemeteredunitcreate.js";
import { ProductPriceSeatBasedCreate$inboundSchema, ProductPriceSeatBasedCreate$outboundSchema, } from "./productpriceseatbasedcreate.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const ProductCreateRecurringMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const ProductCreateRecurringMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateRecurringMetadata$;
(function (ProductCreateRecurringMetadata$) {
    /** @deprecated use `ProductCreateRecurringMetadata$inboundSchema` instead. */
    ProductCreateRecurringMetadata$.inboundSchema = ProductCreateRecurringMetadata$inboundSchema;
    /** @deprecated use `ProductCreateRecurringMetadata$outboundSchema` instead. */
    ProductCreateRecurringMetadata$.outboundSchema = ProductCreateRecurringMetadata$outboundSchema;
})(ProductCreateRecurringMetadata$ || (ProductCreateRecurringMetadata$ = {}));
export function productCreateRecurringMetadataToJSON(productCreateRecurringMetadata) {
    return JSON.stringify(ProductCreateRecurringMetadata$outboundSchema.parse(productCreateRecurringMetadata));
}
export function productCreateRecurringMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateRecurringMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurringMetadata' from JSON`);
}
/** @internal */
export const ProductCreateRecurringPrices$inboundSchema = z.union([
    ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
export const ProductCreateRecurringPrices$outboundSchema = z.union([
    ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateRecurringPrices$;
(function (ProductCreateRecurringPrices$) {
    /** @deprecated use `ProductCreateRecurringPrices$inboundSchema` instead. */
    ProductCreateRecurringPrices$.inboundSchema = ProductCreateRecurringPrices$inboundSchema;
    /** @deprecated use `ProductCreateRecurringPrices$outboundSchema` instead. */
    ProductCreateRecurringPrices$.outboundSchema = ProductCreateRecurringPrices$outboundSchema;
})(ProductCreateRecurringPrices$ || (ProductCreateRecurringPrices$ = {}));
export function productCreateRecurringPricesToJSON(productCreateRecurringPrices) {
    return JSON.stringify(ProductCreateRecurringPrices$outboundSchema.parse(productCreateRecurringPrices));
}
export function productCreateRecurringPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateRecurringPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurringPrices' from JSON`);
}
/** @internal */
export const ProductCreateRecurring$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
    trial_interval: z.nullable(TrialInterval$inboundSchema).optional(),
    trial_interval_count: z.nullable(z.number().int()).optional(),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
export const ProductCreateRecurring$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
    trialInterval: z.nullable(TrialInterval$outboundSchema).optional(),
    trialIntervalCount: z.nullable(z.number().int()).optional(),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductCreateRecurring$;
(function (ProductCreateRecurring$) {
    /** @deprecated use `ProductCreateRecurring$inboundSchema` instead. */
    ProductCreateRecurring$.inboundSchema = ProductCreateRecurring$inboundSchema;
    /** @deprecated use `ProductCreateRecurring$outboundSchema` instead. */
    ProductCreateRecurring$.outboundSchema = ProductCreateRecurring$outboundSchema;
})(ProductCreateRecurring$ || (ProductCreateRecurring$ = {}));
export function productCreateRecurringToJSON(productCreateRecurring) {
    return JSON.stringify(ProductCreateRecurring$outboundSchema.parse(productCreateRecurring));
}
export function productCreateRecurringFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductCreateRecurring$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateRecurring' from JSON`);
}
