// @polar-sh/sdk/dist/esm/models/components/productpriceseatbased.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSeatTiers$inboundSchema, ProductPriceSeatTiers$outboundSchema, } from "./productpriceseattiers.js";
import { ProductPriceType$inboundSchema, ProductPriceType$outboundSchema, } from "./productpricetype.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const ProductPriceSeatBased$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("seat_based"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: ProductPriceType$inboundSchema,
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    price_currency: z.string(),
    seat_tiers: ProductPriceSeatTiers$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "price_currency": "priceCurrency",
        "seat_tiers": "seatTiers",
    });
});
/** @internal */
export const ProductPriceSeatBased$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("seat_based"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: ProductPriceType$outboundSchema,
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    priceCurrency: z.string(),
    seatTiers: ProductPriceSeatTiers$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
        priceCurrency: "price_currency",
        seatTiers: "seat_tiers",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceSeatBased$;
(function (ProductPriceSeatBased$) {
    /** @deprecated use `ProductPriceSeatBased$inboundSchema` instead. */
    ProductPriceSeatBased$.inboundSchema = ProductPriceSeatBased$inboundSchema;
    /** @deprecated use `ProductPriceSeatBased$outboundSchema` instead. */
    ProductPriceSeatBased$.outboundSchema = ProductPriceSeatBased$outboundSchema;
})(ProductPriceSeatBased$ || (ProductPriceSeatBased$ = {}));
export function productPriceSeatBasedToJSON(productPriceSeatBased) {
    return JSON.stringify(ProductPriceSeatBased$outboundSchema.parse(productPriceSeatBased));
}
export function productPriceSeatBasedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatBased$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatBased' from JSON`);
}
