// @polar-sh/sdk/dist/esm/models/components/productpriceseattier.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ProductPriceSeatTier$inboundSchema = z.object({
    min_seats: z.number().int(),
    max_seats: z.nullable(z.number().int()).optional(),
    price_per_seat: z.number().int(),
}).transform((v) => {
    return remap$(v, {
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "price_per_seat": "pricePerSeat",
    });
});
/** @internal */
export const ProductPriceSeatTier$outboundSchema = z.object({
    minSeats: z.number().int(),
    maxSeats: z.nullable(z.number().int()).optional(),
    pricePerSeat: z.number().int(),
}).transform((v) => {
    return remap$(v, {
        minSeats: "min_seats",
        maxSeats: "max_seats",
        pricePerSeat: "price_per_seat",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceSeatTier$;
(function (ProductPriceSeatTier$) {
    /** @deprecated use `ProductPriceSeatTier$inboundSchema` instead. */
    ProductPriceSeatTier$.inboundSchema = ProductPriceSeatTier$inboundSchema;
    /** @deprecated use `ProductPriceSeatTier$outboundSchema` instead. */
    ProductPriceSeatTier$.outboundSchema = ProductPriceSeatTier$outboundSchema;
})(ProductPriceSeatTier$ || (ProductPriceSeatTier$ = {}));
export function productPriceSeatTierToJSON(productPriceSeatTier) {
    return JSON.stringify(ProductPriceSeatTier$outboundSchema.parse(productPriceSeatTier));
}
export function productPriceSeatTierFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatTier$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatTier' from JSON`);
}
