// @polar-sh/sdk/dist/esm/models/components/productpriceseatbasedcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSeatTiers$inboundSchema, ProductPriceSeatTiers$outboundSchema, } from "./productpriceseattiers.js";
/** @internal */
export const ProductPriceSeatBasedCreate$inboundSchema = z.object({
    amount_type: z.literal("seat_based"),
    price_currency: z.string().default("usd"),
    seat_tiers: ProductPriceSeatTiers$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "seat_tiers": "seatTiers",
    });
});
/** @internal */
export const ProductPriceSeatBasedCreate$outboundSchema = z.object({
    amountType: z.literal("seat_based"),
    priceCurrency: z.string().default("usd"),
    seatTiers: ProductPriceSeatTiers$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        seatTiers: "seat_tiers",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceSeatBasedCreate$;
(function (ProductPriceSeatBasedCreate$) {
    /** @deprecated use `ProductPriceSeatBasedCreate$inboundSchema` instead. */
    ProductPriceSeatBasedCreate$.inboundSchema = ProductPriceSeatBasedCreate$inboundSchema;
    /** @deprecated use `ProductPriceSeatBasedCreate$outboundSchema` instead. */
    ProductPriceSeatBasedCreate$.outboundSchema = ProductPriceSeatBasedCreate$outboundSchema;
})(ProductPriceSeatBasedCreate$ || (ProductPriceSeatBasedCreate$ = {}));
export function productPriceSeatBasedCreateToJSON(productPriceSeatBasedCreate) {
    return JSON.stringify(ProductPriceSeatBasedCreate$outboundSchema.parse(productPriceSeatBasedCreate));
}
export function productPriceSeatBasedCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatBasedCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatBasedCreate' from JSON`);
}
