// @polar-sh/sdk/dist/esm/models/components/productpriceseattiers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSeatTier$inboundSchema, ProductPriceSeatTier$outboundSchema, } from "./productpriceseattier.js";
/** @internal */
export const ProductPriceSeatTiers$inboundSchema = z.object({
    tiers: z.array(ProductPriceSeatTier$inboundSchema),
});
/** @internal */
export const ProductPriceSeatTiers$outboundSchema = z.object({
    tiers: z.array(ProductPriceSeatTier$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceSeatTiers$;
(function (ProductPriceSeatTiers$) {
    /** @deprecated use `ProductPriceSeatTiers$inboundSchema` instead. */
    ProductPriceSeatTiers$.inboundSchema = ProductPriceSeatTiers$inboundSchema;
    /** @deprecated use `ProductPriceSeatTiers$outboundSchema` instead. */
    ProductPriceSeatTiers$.outboundSchema = ProductPriceSeatTiers$outboundSchema;
})(ProductPriceSeatTiers$ || (ProductPriceSeatTiers$ = {}));
export function productPriceSeatTiersToJSON(productPriceSeatTiers) {
    return JSON.stringify(ProductPriceSeatTiers$outboundSchema.parse(productPriceSeatTiers));
}
export function productPriceSeatTiersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatTiers$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatTiers' from JSON`);
}
