// @polar-sh/sdk/dist/esm/models/components/productupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { AttachedCustomFieldCreate$inboundSchema, AttachedCustomFieldCreate$outboundSchema, } from "./attachedcustomfieldcreate.js";
import { ExistingProductPrice$inboundSchema, ExistingProductPrice$outboundSchema, } from "./existingproductprice.js";
import { ProductPriceCustomCreate$inboundSchema, ProductPriceCustomCreate$outboundSchema, } from "./productpricecustomcreate.js";
import { ProductPriceFixedCreate$inboundSchema, ProductPriceFixedCreate$outboundSchema, } from "./productpricefixedcreate.js";
import { ProductPriceFreeCreate$inboundSchema, ProductPriceFreeCreate$outboundSchema, } from "./productpricefreecreate.js";
import { ProductPriceMeteredUnitCreate$inboundSchema, ProductPriceMeteredUnitCreate$outboundSchema, } from "./productpricemeteredunitcreate.js";
import { ProductPriceSeatBasedCreate$inboundSchema, ProductPriceSeatBasedCreate$outboundSchema, } from "./productpriceseatbasedcreate.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const ProductUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const ProductUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductUpdateMetadata$;
(function (ProductUpdateMetadata$) {
    /** @deprecated use `ProductUpdateMetadata$inboundSchema` instead. */
    ProductUpdateMetadata$.inboundSchema = ProductUpdateMetadata$inboundSchema;
    /** @deprecated use `ProductUpdateMetadata$outboundSchema` instead. */
    ProductUpdateMetadata$.outboundSchema = ProductUpdateMetadata$outboundSchema;
})(ProductUpdateMetadata$ || (ProductUpdateMetadata$ = {}));
export function productUpdateMetadataToJSON(productUpdateMetadata) {
    return JSON.stringify(ProductUpdateMetadata$outboundSchema.parse(productUpdateMetadata));
}
export function productUpdateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdateMetadata' from JSON`);
}
/** @internal */
export const Two$inboundSchema = z.union([
    ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
export const Two$outboundSchema = z
    .union([
    ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Two$;
(function (Two$) {
    /** @deprecated use `Two$inboundSchema` instead. */
    Two$.inboundSchema = Two$inboundSchema;
    /** @deprecated use `Two$outboundSchema` instead. */
    Two$.outboundSchema = Two$outboundSchema;
})(Two$ || (Two$ = {}));
export function twoToJSON(two) {
    return JSON.stringify(Two$outboundSchema.parse(two));
}
export function twoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Two$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Two' from JSON`);
}
/** @internal */
export const ProductUpdatePrices$inboundSchema = z.union([
    ExistingProductPrice$inboundSchema,
    z.union([
        ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
        ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
            amountType: v.amount_type,
        }))),
    ]),
]);
/** @internal */
export const ProductUpdatePrices$outboundSchema = z.union([
    ExistingProductPrice$outboundSchema,
    z.union([
        ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
        ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
            amount_type: v.amountType,
        }))),
    ]),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductUpdatePrices$;
(function (ProductUpdatePrices$) {
    /** @deprecated use `ProductUpdatePrices$inboundSchema` instead. */
    ProductUpdatePrices$.inboundSchema = ProductUpdatePrices$inboundSchema;
    /** @deprecated use `ProductUpdatePrices$outboundSchema` instead. */
    ProductUpdatePrices$.outboundSchema = ProductUpdatePrices$outboundSchema;
})(ProductUpdatePrices$ || (ProductUpdatePrices$ = {}));
export function productUpdatePricesToJSON(productUpdatePrices) {
    return JSON.stringify(ProductUpdatePrices$outboundSchema.parse(productUpdatePrices));
}
export function productUpdatePricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductUpdatePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdatePrices' from JSON`);
}
/** @internal */
export const ProductUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trial_interval: z.nullable(TrialInterval$inboundSchema).optional(),
    trial_interval_count: z.nullable(z.number().int()).optional(),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema)
        .optional(),
    is_archived: z.nullable(z.boolean()).optional(),
    prices: z.nullable(z.array(z.union([
        ExistingProductPrice$inboundSchema,
        z.union([
            ProductPriceMeteredUnitCreate$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({ amountType: v.amount_type }))),
            ProductPriceFixedCreate$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
            ProductPriceSeatBasedCreate$inboundSchema.and(z.object({ amount_type: z.literal("seat_based") }).transform((v) => ({ amountType: v.amount_type }))),
            ProductPriceCustomCreate$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
            ProductPriceFreeCreate$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
                amountType: v.amount_type,
            }))),
        ]),
    ]))).optional(),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.nullable(z.array(AttachedCustomFieldCreate$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "is_archived": "isArchived",
        "attached_custom_fields": "attachedCustomFields",
    });
});
/** @internal */
export const ProductUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    trialInterval: z.nullable(TrialInterval$outboundSchema).optional(),
    trialIntervalCount: z.nullable(z.number().int()).optional(),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema)
        .optional(),
    isArchived: z.nullable(z.boolean()).optional(),
    prices: z.nullable(z.array(z.union([
        ExistingProductPrice$outboundSchema,
        z.union([
            ProductPriceMeteredUnitCreate$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({ amount_type: v.amountType }))),
            ProductPriceFixedCreate$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
            ProductPriceSeatBasedCreate$outboundSchema.and(z.object({ amountType: z.literal("seat_based") }).transform((v) => ({ amount_type: v.amountType }))),
            ProductPriceCustomCreate$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
            ProductPriceFreeCreate$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
                amount_type: v.amountType,
            }))),
        ]),
    ]))).optional(),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.nullable(z.array(AttachedCustomFieldCreate$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        isArchived: "is_archived",
        attachedCustomFields: "attached_custom_fields",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductUpdate$;
(function (ProductUpdate$) {
    /** @deprecated use `ProductUpdate$inboundSchema` instead. */
    ProductUpdate$.inboundSchema = ProductUpdate$inboundSchema;
    /** @deprecated use `ProductUpdate$outboundSchema` instead. */
    ProductUpdate$.outboundSchema = ProductUpdate$outboundSchema;
})(ProductUpdate$ || (ProductUpdate$ = {}));
export function productUpdateToJSON(productUpdate) {
    return JSON.stringify(ProductUpdate$outboundSchema.parse(productUpdate));
}
export function productUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductUpdate' from JSON`);
}
