// @polar-sh/sdk/dist/esm/models/components/seatslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSeat$inboundSchema, CustomerSeat$outboundSchema, } from "./customerseat.js";
/** @internal */
export const SeatsList$inboundSchema = z.object({
    seats: z.array(CustomerSeat$inboundSchema),
    available_seats: z.number().int(),
    total_seats: z.number().int(),
}).transform((v) => {
    return remap$(v, {
        "available_seats": "availableSeats",
        "total_seats": "totalSeats",
    });
});
/** @internal */
export const SeatsList$outboundSchema = z.object({
    seats: z.array(CustomerSeat$outboundSchema),
    availableSeats: z.number().int(),
    totalSeats: z.number().int(),
}).transform((v) => {
    return remap$(v, {
        availableSeats: "available_seats",
        totalSeats: "total_seats",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SeatsList$;
(function (SeatsList$) {
    /** @deprecated use `SeatsList$inboundSchema` instead. */
    SeatsList$.inboundSchema = SeatsList$inboundSchema;
    /** @deprecated use `SeatsList$outboundSchema` instead. */
    SeatsList$.outboundSchema = SeatsList$outboundSchema;
})(SeatsList$ || (SeatsList$ = {}));
export function seatsListToJSON(seatsList) {
    return JSON.stringify(SeatsList$outboundSchema.parse(seatsList));
}
export function seatsListFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SeatsList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SeatsList' from JSON`);
}
