// @polar-sh/sdk/dist/esm/models/components/seatassign.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SeatAssign$inboundSchema = z.object({
    subscription_id: z.nullable(z.string()).optional(),
    checkout_id: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    external_customer_id: z.nullable(z.string()).optional(),
    customer_id: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "external_customer_id": "externalCustomerId",
        "customer_id": "customerId",
    });
});
/** @internal */
export const SeatAssign$outboundSchema = z.object({
    subscriptionId: z.nullable(z.string()).optional(),
    checkoutId: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    externalCustomerId: z.nullable(z.string()).optional(),
    customerId: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
        checkoutId: "checkout_id",
        externalCustomerId: "external_customer_id",
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SeatAssign$;
(function (SeatAssign$) {
    /** @deprecated use `SeatAssign$inboundSchema` instead. */
    SeatAssign$.inboundSchema = SeatAssign$inboundSchema;
    /** @deprecated use `SeatAssign$outboundSchema` instead. */
    SeatAssign$.outboundSchema = SeatAssign$outboundSchema;
})(SeatAssign$ || (SeatAssign$ = {}));
export function seatAssignToJSON(seatAssign) {
    return JSON.stringify(SeatAssign$outboundSchema.parse(seatAssign));
}
export function seatAssignFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SeatAssign$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SeatAssign' from JSON`);
}
