// @polar-sh/sdk/dist/esm/models/components/subscriptionupdatetrial.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionUpdateTrial$inboundSchema = z.object({
    trial_end: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return remap$(v, {
        "trial_end": "trialEnd",
    });
});
/** @internal */
export const SubscriptionUpdateTrial$outboundSchema = z.object({
    trialEnd: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return remap$(v, {
        trialEnd: "trial_end",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionUpdateTrial$;
(function (SubscriptionUpdateTrial$) {
    /** @deprecated use `SubscriptionUpdateTrial$inboundSchema` instead. */
    SubscriptionUpdateTrial$.inboundSchema = SubscriptionUpdateTrial$inboundSchema;
    /** @deprecated use `SubscriptionUpdateTrial$outboundSchema` instead. */
    SubscriptionUpdateTrial$.outboundSchema = SubscriptionUpdateTrial$outboundSchema;
})(SubscriptionUpdateTrial$ || (SubscriptionUpdateTrial$ = {}));
export function subscriptionUpdateTrialToJSON(subscriptionUpdateTrial) {
    return JSON.stringify(SubscriptionUpdateTrial$outboundSchema.parse(subscriptionUpdateTrial));
}
export function subscriptionUpdateTrialFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionUpdateTrial$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionUpdateTrial' from JSON`);
}
