// @polar-sh/sdk/dist/esm/models/components/webtokenrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
export const WebTokenRequestSubType = {
    User: "user",
    Organization: "organization",
};
/** @internal */
export const WebTokenRequestSubType$inboundSchema = z.nativeEnum(WebTokenRequestSubType);
/** @internal */
export const WebTokenRequestSubType$outboundSchema = WebTokenRequestSubType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebTokenRequestSubType$;
(function (WebTokenRequestSubType$) {
    /** @deprecated use `WebTokenRequestSubType$inboundSchema` instead. */
    WebTokenRequestSubType$.inboundSchema = WebTokenRequestSubType$inboundSchema;
    /** @deprecated use `WebTokenRequestSubType$outboundSchema` instead. */
    WebTokenRequestSubType$.outboundSchema = WebTokenRequestSubType$outboundSchema;
})(WebTokenRequestSubType$ || (WebTokenRequestSubType$ = {}));
/** @internal */
export const WebTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("web"),
    client_id: z.string(),
    client_secret: z.string(),
    session_token: z.string(),
    sub_type: WebTokenRequestSubType$inboundSchema.default("user"),
    sub: z.nullable(z.string()).optional(),
    scope: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "session_token": "sessionToken",
        "sub_type": "subType",
    });
});
/** @internal */
export const WebTokenRequest$outboundSchema = z.object({
    grantType: z.literal("web"),
    clientId: z.string(),
    clientSecret: z.string(),
    sessionToken: z.string(),
    subType: WebTokenRequestSubType$outboundSchema.default("user"),
    sub: z.nullable(z.string()).optional(),
    scope: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        sessionToken: "session_token",
        subType: "sub_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebTokenRequest$;
(function (WebTokenRequest$) {
    /** @deprecated use `WebTokenRequest$inboundSchema` instead. */
    WebTokenRequest$.inboundSchema = WebTokenRequest$inboundSchema;
    /** @deprecated use `WebTokenRequest$outboundSchema` instead. */
    WebTokenRequest$.outboundSchema = WebTokenRequest$outboundSchema;
})(WebTokenRequest$ || (WebTokenRequest$ = {}));
export function webTokenRequestToJSON(webTokenRequest) {
    return JSON.stringify(WebTokenRequest$outboundSchema.parse(webTokenRequest));
}
export function webTokenRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebTokenRequest' from JSON`);
}
